<div class="container">

<table style="width: 100%;"><tr>
<td>Measure.R2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>r^2 measure</h2>

<h3>Description</h3>

<p>This function estimates the usual measure of linkage disequilibrium (<code class="reqn">r^2</code>)  between two loci.</p>


<h3>Usage</h3>

<pre><code class="language-R">Measure.R2(biloci, na.presence = TRUE)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>biloci</code></td>
<td>
<p>Numeric matrix (N x 2), where N is the number of genotypes (or haplotypes)
</p>
<p>Matrix values are the allelic doses:
</p>
<p>- (0,1,2) for genotypes.
</p>
<p>- (0,1) for haplotypes.
</p>
<p>Rows names correspond to the ID of individuals.
</p>
<p>Columns names correspond to the ID of markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.presence</code></td>
<td>
<p>Boolean indicating the presence of missing values in data.
</p>
<p>If na.presence=FALSE (no missing data), computation of <code class="reqn">r^2_V</code> and <code class="reqn">r^2_{VS}</code> is largely optimized.
</p>
<p>By default, na.presence=TRUE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The returned value is the estimated value of the usual measure of linkage disequilibrium (<code class="reqn">r^2</code>) 
or NA if less than 5 individuals have non-missing data at both loci
</p>


<h3>Author(s)</h3>

<p>David Desrousseaux, Florian Sandron, Aur√©lie Siberchicot, Christine Cierco-Ayrolles and Brigitte Mangin</p>


<h3>References</h3>

<p>Hill, W.G, Robertson, A. (1968). Linkage diseqilibrium in finite populations. Theoretical and Applied Genetics, 38, 226-231. DOI: 10.1007/BF01245622</p>


<h3>Examples</h3>

<pre><code class="language-R">data(data.test)
Geno &lt;- data.test[[1]]
Measure.R2(Geno)
</code></pre>


</div>