<div class="container">

<table style="width: 100%;"><tr>
<td>Measure.R2S</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>r^2_S measure</h2>

<h3>Description</h3>

<p>This function estimates the novel measure of linkage disequilibrium which is corrected by the structure of the sample.</p>


<h3>Usage</h3>

<pre><code class="language-R">Measure.R2S(biloci, struc, na.presence=TRUE)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>biloci</code></td>
<td>
<p>Numeric matrix (N x 2), where N is the number of genotypes (or haplotypes)
</p>
<p>Matrix values are the allelic doses:
</p>
<p>- (0,1,2) for genotypes.
</p>
<p>- (0,1) for haplotypes.
</p>
<p>Row names correspond to the  ID of individuals.
</p>
<p>Column names correspond to the ID of markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>struc</code></td>
<td>
<p>Numeric matrix (N x (P-1)), where N is the number of genotypes (or haplotypes) and P the number of sub-populations.
</p>
<p>Matrix values are the probabilities for each genotypes (or haplotypes) to belong to each sub-populations.
</p>
<p>Row names must correspond to the ID of individuals and must be ranged as in the biloci matrix.
</p>
<p>Column names correspond to the ID of sub-populations.
</p>
<p>The matrix must be inversible, if the structure is with P sub-populations, only P-1 columns are expected.
</p>
<p>No missing value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.presence</code></td>
<td>
<p>Boolean indicating the presence of missing values in data.
</p>
<p>If na.presence=FALSE (no missing data), computation of <code class="reqn">r^2_V</code> and <code class="reqn">r^2_{VS}</code> is largely optimized.
</p>
<p>By default, na.presence=TRUE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The returned value is the estimated value of the measure of linkage disequilibrium  corrected by  the structure of the sample or NA if less than 5 individuals have non-missing data at both loci.</p>


<h3>Author(s)</h3>

<p>David Desrousseaux, Florian Sandron, Aur√©lie Siberchicot, Christine Cierco-Ayrolles and Brigitte Mangin</p>


<h3>References</h3>

<p>Mangin, B., Siberchicot, A., Nicolas, S., Doligez, A., This, P., Cierco-Ayrolles, C. (2012). Novel measures of linkage disequilibrium that correct the bias due to population structure and relatedness. Heredity, 108 (3), 285-291. DOI: 10.1038/hdy.2011.73</p>


<h3>Examples</h3>

<pre><code class="language-R">data(data.test)
Geno &lt;- data.test[[1]]
S.2POP &lt;- data.test[[3]]
Measure.R2S(Geno, S.2POP)
</code></pre>


</div>