<div class="container">

<table style="width: 100%;"><tr>
<td>LDexpress</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Determine if genomic variants are associated with gene expression.</h2>

<h3>Description</h3>

<p>Search if a list of genomic variants (or variants in LD with those variants) is associated with gene
expression in tissues of interest. Quantitative trait loci data is downloaded from the
GTEx Portal (<a href="https://gtexportal.org/home/">https://gtexportal.org/home/</a>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">LDexpress(
  snps,
  pop = "CEU",
  tissue = "ALL",
  r2d = "r2",
  r2d_threshold = 0.1,
  p_threshold = 0.1,
  win_size = 5e+05,
  genome_build = "grch37",
  token = NULL,
  file = FALSE,
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>snps</code></td>
<td>
<p>between 1 - 10 variants, using an rsID or chromosome coordinate (e.g. "chr7:24966446")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>a 1000 Genomes Project population, (e.g. YRI or CEU), multiple allowed, default = "CEU".
Use the 'list_pop' function to see a list of available human reference populations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tissue</code></td>
<td>
<p>select from 1 - 54 non-diseased tissue sites collected for the GTEx project, multiple
allowed.  Acceptable user input is taken either from "tissue_name_ldexpress" or "tissue_abbrev_ldexpress"
(tissue abbreviation) code listed in available GTEx tissue sites using the
<code>list_getex_tissues()</code> function (e.g. "ADI_SUB" for Adipose Subcutaneous). Input is case sensitive.
Default = "ALL" for all available tissue types.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r2d</code></td>
<td>
<p>either "r2" for LD R2 or "d" for LD D', default = "r2".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r2d_threshold</code></td>
<td>
<p>R2 or D' (depends on 'r2d' user input parameter) threshold for LD filtering. Any variants
within -/+ of the specified genomic window and R^2 or D' less than the threshold will be removed. Value needs
to be in the range 0 to 1. Default value is 0.1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_threshold</code></td>
<td>
<p>define the eQTL significance threshold used for returning query results. Default value
is 0.1 which returns all GTEx eQTL associations with P-value less than 0.1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>win_size</code></td>
<td>
<p>set genomic window size for LD calculation. Specify a value greater than or equal to zero and less than or
equal to 1,000,000 basepairs (bp). Default value is -/+ 500,000bp.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>Optional character string naming a path and file for saving results.  If file = FALSE, no file will be generated, default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame of all query variant RS numbers, respective QTL which are in LD with query variant,
and associated gene expression.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: LDexpress(snps = c("rs345", "rs456"),
                   pop = c("YRI", "CEU"),
                   tissue = c("ADI_SUB", "ADI_VIS_OME"),
                   r2d = "r2",
                   r2d_threshold = "0.1",
                   p_threshold = "0.1",
                   win_size = "500000",
                   genome_build = "grch37",
                   token = Sys.getenv("LDLINK_TOKEN")
                  )
         
## End(Not run)

</code></pre>


</div>