<div class="container">

<table style="width: 100%;"><tr>
<td>LDpair</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Investigates potentially correlated alleles for a
pair of variants.</h2>

<h3>Description</h3>

<p>Investigates potentially correlated alleles for a
pair of variants.
</p>


<h3>Usage</h3>

<pre><code class="language-R">LDpair(
  var1,
  var2,
  pop = "CEU",
  token = NULL,
  output = "table",
  file = FALSE,
  genome_build = "grch37",
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>var1</code></td>
<td>
<p>the first RS number or genomic coordinate (e.g. "chr7:24966446")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var2</code></td>
<td>
<p>the second RS number or genomic coordinate (e.g. "ch7:24966446")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>a 1000 Genomes Project population(s), (e.g. YRI or CEU), multiple allowed, default = "CEU"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at  <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>two output options available, "text", which displays a two-by-two matrix displaying haplotype counts and allele frequencies along with other statistics, or "table", which displays the same data in rows and columns, default = "table"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>Optional character string naming a path and file for saving results.  If file = FALSE, no file will be generated, default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>text or data frame, depending on the output option
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: LDpair(var1 = "rs3", var2 = "rs4", pop = "YRI", token = Sys.getenv("LDLINK_TOKEN"))
## Not run: LDpair("rs3", "rs4", "YRI", token = Sys.getenv("LDLINK_TOKEN"), "text")

</code></pre>


</div>