<div class="container">

<table style="width: 100%;"><tr>
<td>plot.lddf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot the output of <code>mldest()</code> or
<code>sldest()</code> using <code>corrplot()</code>
</h2>

<h3>Description</h3>

<p>Formats the LD estimates in the form of a matrix and creates a heatmap of
these estimates. This heatmap is created using the
<code>corrplot</code> R package. I've adjusted a lot of the defaults
to suit my visualization preferences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'lddf'
plot(
  x,
  element = "r2",
  type = c("upper", "full", "lower"),
  method = c("color", "circle", "square", "ellipse", "number", "shade", "pie"),
  diag = FALSE,
  is.corr = NULL,
  tl.pos = "n",
  title = NULL,
  na.label = "square",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>lddf</code>, usually created using
either <code>mldest()</code> or <code>sldest()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>element</code></td>
<td>
<p>Which element of <code>x</code> should we plot?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Character, <code>"full"</code>,
<code>"upper"</code> (default) or <code>"lower"</code>, display
full matrix, lower triangular or upper
triangular matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>See <code>corrplot()</code> for available options.
Default value is <code>"color"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diag</code></td>
<td>
<p>Logical, whether display the correlation coefficients
on the principal diagonal.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.corr</code></td>
<td>
<p>See <code>corrplot()</code>. Default behavior
is <code>TRUE</code> if an element is constrained
between -1 and 1 and <code>FALSE</code> otherwise.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tl.pos</code></td>
<td>
<p>See <code>corrplot()</code>. Default value
is <code>"n"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>What should the title be? Defaults to the element name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.label</code></td>
<td>
<p>See <code>corrplot()</code>. Default value
is <code>"square"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments to pass to
<code>corrplot()</code>. See the documentation of that
function for options.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For greater plotting flexibility, see <code>corrplot()</code>
for the parameter options.
</p>


<h3>Value</h3>

<p>(Invisibly) returns a matrix of the selected elements.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(1)

## Simulate genotypes when true correlation is 0
nloci &lt;- 5
nind  &lt;- 100
K &lt;- 6
nc &lt;- 1
genomat &lt;- matrix(sample(0:K, nind * nloci, TRUE), nrow = nloci)

## Haplotypic LD estimates
lddf &lt;- mldest(geno = genomat,
               K = K,
               nc = nc,
               type = "hap")

## Plot estimates of z
plot(lddf, element = "z")

</code></pre>


</div>