<div class="container">

<table style="width: 100%;"><tr>
<td>Dprime</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get the standardized composite D'.</h2>

<h3>Description</h3>

<p>This function will either standardize by the maximum covariance
conditional on the marginal genotype distribution, or by the
maximum covariance conditional on the marginal allele frequencies.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Dprime(qmat, type = c("allele", "geno"), constrain = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>qmat</code></td>
<td>
<p>The observed joint genotype distribution.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Should we condition on the marginal genotype distribution
(<code>type = "geno"</code>), or should we condition on the allele frequency
(<code>type = "allele"</code>)?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>constrain</code></td>
<td>
<p>A logical. This option is only applicable when
<code>type = "allele"</code>. Should return an value that is equal
to D' under HWE (<code>FALSE</code>) or a value that is constrained
to lie between -1 and 1 (<code>TRUE</code>)? Defaults to <code>FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that when <code>type = "allele"</code> and <code>constrain = FALSE</code>,
the resulting D' is constrained to fall between -K and K, where
K is the ploidy of the species. However, under HWE, this measure is
equal to haplotypic D'. Using <code>constrain = TRUE</code> will result
in a measure that is constrained to lie between -1 and 1, but
it will not equal haplotypic D' under HWE.
</p>
<p>Using <code>type = "geno"</code> is its own thing and will not equal
D' generally under HWE. When <code>type = "geno"</code>, then the
the <code>constrain</code> parameter has no effect.
</p>


<h3>Value</h3>

<p>A vector of length 2. The first element is the estimated
D'. The second element is the normalization used.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class="language-R">K &lt;- 6
qmat &lt;- matrix(stats::runif((K+1)^2), nrow = K+1)
qmat &lt;- qmat / sum(qmat)
Dprime(qmat, type = "geno")
Dprime(qmat, type = "allele")

</code></pre>


</div>