<div class="container">

<table style="width: 100%;"><tr>
<td>ldsep-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Linkage Disequilibrium Shrinkage Estimation for Polyploids</h2>

<h3>Description</h3>

<p>Estimate haplotypic or composite pairwise linkage disequilibrium
(LD) in polyploids, using either genotypes or genotype likelihoods. Support is
provided to estimate the popular measures of LD: the LD coefficient D,
the standardized LD coefficient D', and the Pearson correlation
coefficient r. All estimates are returned with corresponding
standard errors. These estimates and standard errors can then be used
for shrinkage estimation.
</p>


<h3>Functions</h3>

<p>The main functions are:
</p>

<dl>
<dt><code>ldfast()</code></dt>
<dd>
<p>Fast, moment-based, bias-corrected LD
LD estimates from marginal posterior distributions.</p>
</dd>
<dt><code>ldest()</code></dt>
<dd>
<p>Estimates pairwise LD.</p>
</dd>
<dt><code>mldest()</code></dt>
<dd>
<p>Iteratively apply <code>ldest()</code>
across many pairs of SNPs.</p>
</dd>
<dt><code>sldest()</code></dt>
<dd>
<p>Iteratively apply <code>ldest()</code>
along a sliding window of fixed length.</p>
</dd>
<dt><code>plot.lddf()</code></dt>
<dd>
<p>Plot method for the output of
<code>mldest()</code> and <code>sldest()</code>.</p>
</dd>
<dt><code>format_lddf()</code></dt>
<dd>
<p>Format the output of
<code>mldest()</code> and <code>sldest()</code> into a matrix.</p>
</dd>
<dt><code>ldshrink()</code></dt>
<dd>
<p>Shrink correlation estimates
using adaptive shrinkage (Stephens, 2017; Dey and Stephens, 2018).</p>
</dd>
</dl>
<h3>Citation</h3>

<p>If you find the methods in this package useful, please run the following
in R for citation information: <code>citation("ldsep")</code>
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


</div>