<div class="container">

<table style="width: 100%;"><tr>
<td>format_lddf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Format an element of <code>mldest()</code> or
<code>sldest()</code> into an
upper-triangular matrix.</h2>

<h3>Description</h3>

<p>Formats the LD estimates and standard errors output
from running <code>mldest()</code> or <code>sldest()</code>
into a more conventional upper-triangular matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">format_lddf(obj, element = "r2")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>An object of class <code>lddf</code>, usually output from
running either <code>mldest()</code> or
<code>sldest()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>element</code></td>
<td>
<p>Which element in <code>obj</code> should we format into an
upper-triangular matrix?</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix of the selected elements. Only the upper-triangle of the
matrix is filled. The lower-triangle and the diagonal are <code>NA</code>'s.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(1)

## Simulate genotypes when true correlation is 0
nloci &lt;- 5
nind  &lt;- 100
K &lt;- 6
nc &lt;- 1
genomat &lt;- matrix(sample(0:K, nind * nloci, TRUE), nrow = nloci)

## Haplotypic LD estimates
lddf &lt;- mldest(geno = genomat,
               K = K,
               nc = nc,
               type = "hap")

## Obtain the D estimates in matrix form
Dmat &lt;- format_lddf(obj = lddf, element = "D")
Dmat

</code></pre>


</div>