<div class="container">

<table style="width: 100%;"><tr>
<td>msd</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Migration, drift, and selection</h2>

<h3>Description</h3>

<p>Simulates migration, natural selection, and genetic drift. Selection can be in opposite directions in the two populations experiencing gene flow.
</p>


<h3>Usage</h3>

<pre><code class="language-R">	msd(p0=c(0.5,0.5), Ne=c(100,100), w=list(c(1,1,1),c(1,1,1)), 
		m=c(0.01,0.01), ngen=400, colors=c("red","blue"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p0</code></td>
<td>
<p>starting frequency for the <em>A</em> allele in each of two populations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Ne</code></td>
<td>
<p>effective population size for each of two populations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>w</code></td>
<td>
<p>fitnesses of the three genotypes (<em>AA</em>, <em>Aa</em>, and <em>aa</em>, in that order) in each of the two populations. <code>w</code> should take the form of a list of two vectors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>rates of migration <em>from</em> the first population to the second, and from the second population to the first, in that order. This value is best interpreted as the <em>probability</em> that an individual born in population 1 will migrate to population 2 before reproduction, and <em>vice versa</em>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ngen</code></td>
<td>
<p>total time, in number of generations, for the simulation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colors</code></td>
<td>
<p>colors to use for plotting.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>optional arguments. Presently, the only optional argument for <code>msd</code> is <code>show.legend</code> (which defaults to <code>TRUE</code>). The <code>plot</code> method adds the additional optional arguments of <code>colors</code> (a vector of colors for the two simulated populations), <code>lwd</code>, and <code>type</code> (e.g., <code>"l"</code> or <code>"s"</code>).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The function creates a plot and invisibly returns a list containing the allele frequency through time for each of the two simulated populations.
</p>
<p>The returned object is of class <code>"msd"</code> and can be printed or re-plotted using corresponding <code>print</code> or <code>plot</code> methods. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code>drift.selection</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">	msd()
	msd(p0=c(0.25,0.75),w=list(c(1,0.9,0.8),c(0.8,0.9,1)))
	object&lt;-msd(p0=c(0.75,0.25),w=list(c(1,0.9,0.8),
		c(0.8,0.9,1)),m=c(0.1,0.1),ngen=100)
	print(object)
	plot(object,colors=c("black","grey"),lwd=4,type="s")
</code></pre>


</div>