<div class="container">

<table style="width: 100%;"><tr>
<td>LFDR.MM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Performs a Multiple Hypothesis Testing Using the Method of Moments</h2>

<h3>Description</h3>

<p>Based on a given vector of chi-square test statistics, provides estimates of local false discoveries.
</p>


<h3>Usage</h3>

<pre><code class="language-R">LFDR.MM(x)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A vector of chi-square test statistics with one degree of freedom.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>For <code class="reqn">N</code> given features (genes, proteins, SNPs, etc.), the function tests the null hypothesis <code class="reqn">H_{0i}</code>, <code class="reqn">i=1,\ldots,N</code>, indicating that there is no association between feature <code class="reqn">i</code> and a specific disease, versus its alternative hypothesis <code class="reqn">H_{1i}</code>. For each unassociated feature <code class="reqn">i</code>, it is suppoed that the corresponding test stiatistic <code class="reqn">x_i</code> follows a central chi-square distribution with one degree of freedom.  For each associated feature <code class="reqn">i</code>, it is assumed that the corresponding test stiatistic <code class="reqn">x_i</code> follows a non-central chi-square distribution with one degree of freedom and non-centrality parameter <code class="reqn">\lambda</code>. In this packag, association is measured by estimating the local false discovery rate (LFDR), the posterior probability that the null hypothesis <code class="reqn">H_{0i}</code> given the test statistic <code class="reqn">x_i</code> is true.
This package returns three components as mentioned in the <b>Value</b> section.</p>


<h3>Value</h3>

<p>Outputs three elements as seen below:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>pi0.hat</code></td>
<td>
<p>estimate of proportion of unassocaited features <code class="reqn">\pi_0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncp.hat</code></td>
<td>
<p>estimate of the non-centrality parameter <code class="reqn">\lambda</code> of the chi-square model for associated features.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lfdr.hat</code></td>
<td>
<p>estimates of local false discovery rates.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Code: Ali Karimnezhad. <br>
Documentation: Ali Karimnezhad. <br></p>


<h3>References</h3>

<p>Karimnezhad, A. (2020). A Simple Yet Efficient Parametric Method of Local False Discovery Rate Estimation Designed for Genome-Wide Association Data Analysis. Retrieved from
<a href="https://arxiv.org/abs/1909.13307">https://arxiv.org/abs/1909.13307</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># vector of test statistics for assocaited features
stat.assoc&lt;- rchisq(n=1000,df=1, ncp = 3)

# vector of test statistics for unassocaited features
stat.unassoc&lt;- rchisq(n=9000,df=1, ncp = 0)

# vector of test statistics
stat&lt;- c(stat.assoc,stat.unassoc)

output &lt;- LFDR.MM(x=stat)

# Estimated pi0
output$p0.hat

# Estimated non-centrality parameter
output$ncp.hat

# Estimated LFDRs
output$lfdr.hat
</code></pre>


</div>