<div class="container">

<table style="width: 100%;"><tr>
<td>caution.parameter.actions</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Based on a Decision-Theoretic Approach, Performs a Multiple Hypothesis Testing Problem under a Zero-One Loss Function</h2>

<h3>Description</h3>

<p>Assuming a zero-onr loss function, it provides three caution-type actions using estimated LFDRs computed based on both separate and combined reference classes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">caution.parameter.actions(x1,x2,l1=4,l2=1) # default values l1=4 and l2=1
# to obtain a threshold of 20%.
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x1</code></td>
<td>
<p>A vector of LFDRs in the combined reference class.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x2</code></td>
<td>
<p>A vector of LFDRs in the separate reference class.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>l1</code></td>
<td>
<p>Loss value (Type-I error) for deriving the threshold of the Bayes
action.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>l2</code></td>
<td>
<p>Loss value (Type-II error) for deriving the threshold of the Bayes
action.</p>
</td>
</tr>
</table>
<h3>Details</h3>


<p>Accepts previously obtained LFDR estimates of SNPs falling inside the intersection of the separate and combined reference classes. The LFDR estimates of some biological feature (SNP or gene) within a sample population that we will refer to as ‘reference class’. If a reference class, containing LFDR estimates
is a subset of the other, it is referred to as ‘separate class’.
The entire set of LFDR estimates is called a ‘combined’ reference class. Then,
a multiple hypothesis problem is conducted using three caution-type estimators.
The threshold set for rejecting the null hypothesis is derived from
pre-specified <code>l1</code> and <code>l2</code> values. Since having a type-I error is
worse than a type-II error, <code>l1</code> is recommende to be greater than
<code>l2</code>.
</p>
<p>In generating the output, there are two potential outputs for each index of the
three caution-type actions. Check the <b>Value</b> section for the
corresponding caution-type actions.
</p>
<p>For each index of the output, one of two potential outputs based on Bayes action
are shown:
</p>

<table>
<tr>
<td style="text-align: right;">
 <code>0</code> </td>
<td style="text-align: left;"> Do not reject the null hypothesis </td>
</tr>
<tr>
<td style="text-align: right;">
 <code>1</code> </td>
<td style="text-align: left;"> Reject the null hypothesis</td>
</tr>
<tr>
<td style="text-align: right;">
</td>
</tr>
</table>
<p>For each corresponding index in the output, the decision on whether to reject or
not reject the null hypothesis for biological feature can be based on
<code>CGM1</code>, <code>CGM0</code>, and <code>CGM0.5</code> decisions. Check <b>See Also</b> for
more details on how to better interpret the outputs.
</p>


<h3>Value</h3>

<p>Outputs three vectors of equal size as seen below:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>CGM1</code></td>
<td>
<p>Decision values for the Conditional Gamma Minimax (CGMinimax).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CGM0</code></td>
<td>
<p>Decision values for the Conditional Gamma Minimin (CGMinimin).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CGM0.5</code></td>
<td>
<p>Decision values for the CG0.5 caution case (a balance between
CGMinimax and CGMinimin.</p>
</td>
</tr>
</table>
<p>Note that the length of the input vectors <code>x1</code> and <code>x2</code> determines the
number of null hypothesis values seen in the output.
</p>


<h3>Note</h3>

<p>A limitation to the code is that both reference classes: <code>x1</code> and <code>x2</code>
must be of the same vector length.
</p>


<h3>Author(s)</h3>

<p>Code: Ali Karimnezhad. <br>
Documentation: Justin Chitpin, Anna Akpawu and Johnary Kim. <br></p>


<h3>References</h3>

<p>Karimnezhad, A. and Bickel, D. R. (2016). Incorporating prior knowledge about
genetic variants into the analysis of genetic association data: An empirical
Bayes approach. Working paper. Retrieved from
<a href="http://hdl.handle.net/10393/34889">http://hdl.handle.net/10393/34889</a>
</p>


<h3>See Also</h3>

<p>For more information on how to interpret the outputs, look at the vignette,
“Using <code>LFDREmpiricalBayes</code>”.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#LFDR reference class values generated

#First reference class (separate class)
LFDR.Separate &lt;- c(.14,.8,.251,.30)
#Second reference class (combined class)
LFDR.Combined &lt;- c(.21,.61,.0888,.10)

# Default threshold at (20%).
output &lt;- caution.parameter.actions(x1=LFDR.Separate, x2=LFDR.Combined)

# Three caution cases
output

</code></pre>


</div>