<div class="container">

<table style="width: 100%;"><tr>
<td>lfmm_lasso</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>LFMM least-squares estimates with lasso penalty (Sparse LFMM)</h2>

<h3>Description</h3>

<p>This function computes regularized least squares estimates
for latent factor mixed models using a lasso penalty.
</p>


<h3>Usage</h3>

<pre><code class="language-R">lfmm_lasso(
  Y,
  X,
  K,
  nozero.prop = 0.01,
  mu.num = 100,
  mu.min.ratio = 0.01,
  mu = NULL,
  it.max = 100,
  relative.err.epsilon = 1e-06
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>a response variable matrix with n rows and p columns.
Each column is a response variable (e.g., SNP genotype,
gene expression level, beta-normalized methylation profile, etc).
Response variables must be encoded as numeric.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>an explanatory variable matrix with n rows and d columns.
Each column corresponds to a distinct explanatory variable (eg. phenotype, exposure, outcome).
Explanatory variables must be encoded as numeric.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>an integer for the number of latent factors in the regression model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nozero.prop</code></td>
<td>
<p>a numeric value for the expected proportion of non-zero effect sizes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu.num</code></td>
<td>
<p>a numeric value for the number of 'mu' values (advance parameter).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu.min.ratio</code></td>
<td>
<p>(advance parameter) A fraction of <code>mu.max</code>, the data derived entry value
(i.e. the smallest value for which all coefficients are zero).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>(advance parameter) Smallest value of <code>mu</code>. Null value by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>it.max</code></td>
<td>
<p>an integer value for the number of iterations of the algorithm.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>relative.err.epsilon</code></td>
<td>
<p>a numeric value for a relative convergence error. Determine
whether the algorithm converges or not.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The algorithm minimizes the following penalized least-squares criterion
</p>
<p>The response variable matrix Y and the explanatory variable are centered.
</p>


<h3>Value</h3>

<p>an object of class <code>lfmm</code> with the following attributes:
</p>

<ul>
<li>
<p> U the latent variable score matrix with dimensions n x K,
</p>
</li>
<li>
<p> V the latent variable axes matrix with dimensions p x K,
</p>
</li>
<li>
<p> B the effect size matrix with dimensions p x d.
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Kevin Caye, Basile Jumentier, Olivier Francois
</p>


<h3>References</h3>

<p>B. Jumentier, Caye, K., J. Lepeule, and O. Fran√ßois, 2019 Sparse latent factor regression models for genome-wide and epigenome-wide association studies (in prep)
</p>


<h3>Examples</h3>

<pre><code class="language-R">
library(lfmm)

## An EWAS example with Y = methylation data 
## and X = exposure

## Simulate the data 

dat &lt;- lfmm_sampler(n = 100, 
                    p = 1000,
                    K = 3,
                    outlier.prop = 0.02,
                    cs = 0.1,
                    sigma = 0.2,
                    B.sd = 5,
                    B.mean = 0,
                    U.sd = 1.0,
                    V.sd = 1.0)

Y &lt;- scale(dat$Y)
X &lt;- scale(dat$X)

## Fit an LFMM with 2 latent factors
mod.lfmm &lt;- lfmm_lasso(Y = Y,
                       X = X, 
                       K = 3,
                       nozero.prop = 0.02)
                       
## Manhattan plot of sparse effect sizes
effect &lt;- mod.lfmm$B
causal &lt;- dat$outlier

plot(effect, 
     pch = 19, 
     cex = .3,
     xlab = "Probe",
     col = "grey")
     
points(causal, 
       effect[causal], 
       col = "blue")
</code></pre>


</div>