<div class="container">

<table style="width: 100%;"><tr>
<td>GA.SSD.All</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Genetic association tests for multiple regions/genes using SSD format files
</h2>

<h3>Description</h3>

<p>Test the association between an quantitative outcome and multiple region/genes using SSD format files.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GA.SSD.All(SSD.INFO, result.prelim, Gsub.id=NULL, B=5000, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SSD.INFO</code></td>
<td>

<p>SSD format information file, output of function â€œOpen_SSD". The sets are defined by this file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>result.prelim</code></td>
<td>

<p>Output of function "GA.prelim()".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X, E and time.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>

<p>Number of Bootstrap replicates. The default is 5000.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Other options of the generalized score type test. Defined same as in function "GA.test()".
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Please see SKAT vignettes for using SSD files.

</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>results</code></td>
<td>

<p>Results of the set based analysis. First column contains the set ID; Second column (second and third columns when the MinP test is compared) contains the p-values; Last column contains the number of tested SNPs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>results.single</code></td>
<td>

<p>Results of the single variant analysis for all variants in the sets. First column contains the regions' names; Second column is the variants' names; Third column contains the minor allele frequencies; Last column contains the p.values.
</p>
</td>
</tr>
</table>
</div>