<div class="container">

<table style="width: 100%;"><tr>
<td>GA.test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Test the association between an quantitative outcome variable and a region/gene by a generalized score type test.
</h2>

<h3>Description</h3>

<p>Once the preliminary work is done using "GA.prelim()", this function tests a specifc region/gene. Single SNP analyses are also incorporated.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GA.test(result.prelim,G,Gsub.id=NULL,weights='beta',B=5000, B.coef=NULL,
impute.method='fixed')
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>result.prelim</code></td>
<td>

<p>The output of function "GEI.prelim()"
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>

<p>Genetic variants in the target region/gene, an m*q matrix where m is the subject ID and q is the total number of genetic variables. Note that the number of rows in Z should be same as the number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X and time.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>

<p>Can be a numeric vector of weights for genetic variants (The length should be same as the number of genetic variants in the set.), or pre-determined weights: "beta" (beta weights as in SKAT paper), "rare" (restrited to MAF&lt;0.01), "common" (restrited to MAF&gt;0.01). The default is NULL, where the flat weights are applied.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>

<p>Number of Bootstrap replicates. The default is 5000.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B.coef</code></td>
<td>

<p>Direct import of Bootstrap coefficients, an m by B matrix. This is in order to efficiently implement the Bootstrap step. The default is NULL.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be "random", "fixed" or "bestguess". Given the estimated allele frequency, "random" simulates the genotype from binomial distribution; "fixed" uses the genotype expectation; "Best guess" uses the genotype with highest probability.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>

<p>P-value of the set based generalized score type test.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.single</code></td>
<td>

<p>P-values of the incorporated single SNP analyses
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.marker</code></td>
<td>

<p>number of heterozygous SNPs in the SNP set.
</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">## GA.prelim does the preliminary data management.
# Input: Y, time, X (covariates)
## GA.test tests a region.
# Input: G (genetic variants) and result of GEI.prelim

library(LGEWIS)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by p matrix
# time: describe longitudinal structure, n by 2 matrix
# G: genotype matrix, m by q matrix where m is the total number of subjects

data(LGEWIS.example)
Y&lt;-LGEWIS.example$Y
X&lt;-LGEWIS.example$X
time&lt;-LGEWIS.example$time
G&lt;-LGEWIS.example$G

# Preliminary data management
result.prelim&lt;-GA.prelim(Y,time,X=X)

# test with 5000 bootstrap replicates
result&lt;-GA.test(result.prelim,G,B=5000)
</code></pre>


</div>