<div class="container">

<table style="width: 100%;"><tr>
<td>GEI.SSD.All</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Gene-environment interaction tests for multiple regions/genes using SSD format files
</h2>

<h3>Description</h3>

<p>Test the interaction between an environmental exposure and multiple region/genes on a quantitative outcome using SSD format files.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GEI.SSD.All(SSD.INFO, result.prelim, Gsub.id=NULL, MinP.adjust=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SSD.INFO</code></td>
<td>

<p>SSD format information file, output of function â€œOpen_SSD". The sets are defined by this file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>result.prelim</code></td>
<td>

<p>Output of function "GEI.prelim()".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X, E and time.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MinP.adjust</code></td>
<td>

<p>If the users would like to compare with the MinP test, this parameter specify the adjustment thereshold as in Gao, et al. (2008) "A multiple testing correction method for genetic association studies using correlated single nucleotide polymorphisms". Values from 0 to 1 are permitted. The default is NULL, i.e., no comparison. The value suggested by Gao, et al. (2008) is 0.95.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Other options of the generalized score type test. Defined same as in function "GEI.test()".
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Please see SKAT vignettes for using SSD files.

</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>results</code></td>
<td>

<p>Results of the set based analysis. First column contains the set ID; Second column (second and third columns when the MinP test is compared) contains the p-values; Last column contains the number of tested SNPs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>results.single</code></td>
<td>

<p>Results of the single variant analysis for all variants in the sets. First column contains the regions' names; Second column is the variants' names; Third column contains the minor allele frequencies; Last column contains the p.values.
</p>
</td>
</tr>
</table>
</div>