<div class="container">

<table style="width: 100%;"><tr>
<td>LM_eukaryotes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Transformation in a LifemapR format of NCBI information for 1000 eukaryotes</h2>

<h3>Description</h3>

<p>A dataset containing NCBI information of 1000 eukaryotes. The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(LM_eukaryotes)
</code></pre>


<h3>Format</h3>

<p>A lifemap object - a list containing the basemap used to fetch data and df, a data frame with 2760 rows and 26 variables:
</p>

<dl>
<dt>X.Organism.Name</dt>
<dd>
<p>Organism name at the species level</p>
</dd>
<dt>taxid</dt>
<dd>
<p>NCBI taxid</p>
</dd>
<dt>BioProject.Accession</dt>
<dd>
<p>BioProject Accession number (from BioProject database)</p>
</dd>
<dt>BioProject.ID</dt>
<dd>
<p>BioProject ID</p>
</dd>
<dt>Group</dt>
<dd>
<p>Commonly used organism groups:  Animals, Fungi, Plants, Protists</p>
</dd>
<dt>SubGroup</dt>
<dd>
<p>NCBI Taxonomy level below group: Mammals, Birds, Fishes, Flatworms, Insects, Amphibians, Reptiles, Roundworms, Ascomycetes, Basidiomycetes, Land Plants, Green Algae, Apicomplexans, Kinetoplasts</p>
</dd>
<dt>Size..Mb.</dt>
<dd>
<p>Total length of DNA submitted for the project</p>
</dd>
<dt>GC.</dt>
<dd>
<p>Percent of nitrogenous bases (guanine or cytosine) in DNA submitted for the project</p>
</dd>
<dt>Assembly.Accession</dt>
<dd>
<p>Name of the genome assembly (from NCBI Assembly database)</p>
</dd>
<dt>Replicons</dt>
<dd>
<p>Number of replicons in the assembly</p>
</dd>
<dt>WGS</dt>
<dd>
<p>Four-letter Accession prefix followed by version as defined in WGS division of GenBank/INSDC</p>
</dd>
<dt>Scaffolds</dt>
<dd>
<p>Number of scaffolds in the assembly</p>
</dd>
<dt>Genes</dt>
<dd>
<p>Number of Genes annotated in the assembly</p>
</dd>
<dt>Proteins</dt>
<dd>
<p>Number of Proteins annotated in the assembly</p>
</dd>
<dt>Release.Date</dt>
<dd>
<p>First public sequence release for the project</p>
</dd>
<dt>Modify.Date</dt>
<dd>
<p>Sequence modification date for the project</p>
</dd>
<dt>Status</dt>
<dd>
<p>Highest level of assembly: <br> Chromosomes: one or more chromosomes are assembled<br> Scaffolds or contigs: sequence assembled but no chromosomes</p>
</dd>
<dt>Center</dt>
<dd>
<p>Origin of the sample</p>
</dd>
<dt>BioSample.Accession</dt>
<dd>
<p>BioSample Accession number</p>
</dd>
<dt>lon</dt>
<dd>
<p>longitude of taxids on a specific basemap</p>
</dd>
<dt>lat</dt>
<dd>
<p>latitude of taxids on a specific basemap</p>
</dd>
<dt>sci_name</dt>
<dd>
<p>scientific name of taxids</p>
</dd>
<dt>zoom</dt>
<dd>
<p>zoom of taxids on a specific basemap</p>
</dd>
<dt>ascend</dt>
<dd>
<p>the list of all ancestors of taxids on a specific basemap</p>
</dd>
<dt>type</dt>
<dd>
<p>either "requested" if the taxid was given, "ancestor" if gotten from the database</p>
</dd>
<dt>ancestor</dt>
<dd>
<p>the direct ancestor oftaxids on a specific basemap</p>
</dd>
</dl>
</div>