<div class="container">

<table style="width: 100%;"><tr>
<td>find.gene.pseudomarker</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find nearest peudomarker to each gene</h2>

<h3>Description</h3>

<p>Pull out the pseudomarker that is closest to the position of each of a
series of genes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">find.gene.pseudomarker(cross, pmap, geneloc, where = c("prob", "draws"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>An object of class <code>"cross"</code> containing data for a QTL
experiment.  See the help file for <code>qtl::read.cross()</code> in the
R/qtl package (<a href="https://rqtl.org">https://rqtl.org</a>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pmap</code></td>
<td>
<p>A physical map of the markers in <code>cross</code>, with locations in
Mbp.  This is a list whose components are the marker locations on each
chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneloc</code></td>
<td>
<p>A data frame specifying the physical locations of the genes.
There should be two columns, <code>chr</code> for chromosome and <code>pos</code> for
position in Mbp.  The rownames should indicate the gene names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>where</code></td>
<td>
<p>Indicates whether to pull pseudomarkers from the genotype
probabilities (produced by <code>qtl::calc.genoprob()</code>) or from the
imputed genotypes (produced by <code>qtl::sim.geno()</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>We first convert positions (by interpolation) from those contained within
<code>cross</code> to physical coordinates contained in <code>pmap</code>.  We then use
<code>qtl::find.pseudomarker()</code> to identify the closest pseudomarker to
each gene location.
</p>
<p>We also include the positions of the pseudomarkers, and we print a warning
message if pseudomarkers are &gt; 2 Mbp from the respective gene.
</p>


<h3>Value</h3>

<p>A data frame with columns <code>chr</code> (the chromosome) and
<code>pmark</code> (the name of the pseudomarker).  The third column <code>pos</code>
contains the Mbp position of the pseudomarker.  The final column is the
signed distance between the gene and the pseudomarker.  The rownames
indicate the gene names.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>See Also</h3>

<p><code>qtl::find.pseudomarker()</code>,
<code>qtl::find.pseudomarkerpos()</code>, <code>plotEGclass()</code>,
<code>disteg()</code>, <code>calc.locallod()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(f2cross, expr1, genepos, pmap)
library(qtl)

# calc QTL genotype probabilities
f2cross &lt;- calc.genoprob(f2cross, step=1)

# find nearest pseudomarkers
pmark &lt;- find.gene.pseudomarker(f2cross, pmap, genepos, "prob")

</code></pre>


</div>