<div class="container">

<table style="width: 100%;"><tr>
<td>DT_cornhybrids</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Corn crosses and markers
</h2>

<h3>Description</h3>

<p>This dataset contains phenotpic data for plant height and grain yield for 100 out of 400 possible hybrids originated from 40 inbred lines belonging to 2 heterotic groups, 20 lines in each, 1600 rows exist for the 400 possible hybrids evaluated in 4 locations but only 100 crosses have phenotypic information.  The purpose of this data is to show how to predict the other 300 crosses.
</p>
<p>The data contains 3 elements. The first is the phenotypic data and the parent information for each cross evaluated in the 4 locations. 1200 rows should have missing data but the 100 crosses performed were chosen to be able to estimate the GCA and SCA effects of everything.
</p>
<p>The second element of the data set is the phenotypic data and other relevant information for the 40.
</p>
<p>The third element is the genomic relationship matrix for the 40 inbred lines originated from 511 SNP markers and calculated using the A.mat function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("DT_cornhybrids")</code></pre>


<h3>Format</h3>

<p>The format is:
chr "DT_cornhybrids"
</p>


<h3>Source</h3>

<p>This data was generated by a corn study.
</p>


<h3>References</h3>

<p>Giovanny Covarrubias-Pazaran (2024).  lme4breeding: enabling genetic evaluation in the age of genomic data. To be submitted to Bioinformatics.
</p>
<p>Douglas Bates, Martin Maechler, Ben Bolker, Steve Walker (2015). Fitting Linear Mixed-Effects Models Using lme4. Journal of Statistical Software, 67(1), 1-48.
</p>


<h3>See Also</h3>

<p>The core function of the package <code>lmebreed</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">
data(DT_cornhybrids)
DT &lt;- DT_cornhybrids
GT &lt;- GT_cornhybrids
A &lt;- GT
K1 &lt;- A[levels(DT$GCA1), levels(DT$GCA1)]; dim(K1)
K2 &lt;- A[levels(DT$GCA2), levels(DT$GCA2)]; dim(K2)


S &lt;- kronecker(K1, K2) ; dim(S)
rownames(S) &lt;- colnames(S) &lt;- levels(DT$SCA)

ans &lt;- lmebreed(Yield ~ Location + (1| GCA1) + (1|GCA2),
                relmat = list(GCA1=K1,
                              GCA2=K2),
                data=DT)
vc &lt;- VarCorr(ans); print(vc,comp=c("Variance"))



</code></pre>


</div>