<div class="container">

<table style="width: 100%;"><tr>
<td>lncDIFF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>lncRNA Differential Expression (DE) analysis</h2>

<h3>Description</h3>

<p>lncDIFF returns DE analysis results based on lncRNA counts and grouping variables.
</p>


<h3>Usage</h3>

<pre><code class="language-R">lncDIFF(
  edata,
  group,
  covariate = NULL,
  link.function = "log",
  CompareGroups = NULL,
  simulated.pvalue = FALSE,
  permutation = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>edata</code></td>
<td>
<p>Normalized counts matrix with genes in rows and samples in columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group</code></td>
<td>
<p>Primary factor of interest in DE analysis, e.g., treatment groups, tissue types, other phenotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covariate</code></td>
<td>
<p>Other variables (or covariates) associated with expression level. Input must be a matrix or data frame with each column being a covariate matching to <code>group</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>link.function</code></td>
<td>
<p>Link function for the generalized linear model, either 'log' or 'identity', default as 'log'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CompareGroups</code></td>
<td>
<p>Labels of treatment groups or phenotypes of interest to be compared in DE analysis. Input must be a vector of <code>group</code> labels without duplicates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simulated.pvalue</code></td>
<td>
<p>If empirical p-values are computed, simulated.pvalue=TRUE. The default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>permutation</code></td>
<td>
<p>The number of permutations used in simulating pvalues. The default value is 100.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>DE.results</code></td>
<td>
<p>Likelihood ratio test results with test statistics, p-value, FDR, DE genes, groupwise mean expression, fold change (if two groups are compared). 
If simulated.pvalue=TRUE, test.results also includes simulated p-value and FDR.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>full.model.fit</code></td>
<td>
<p>Generalized linear model with zero-inflated Exponential likelihood function, estimating group effect compared to a reference group.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Li, Q., Yu, X., Chaudhary, R. et al.'lncDIFF: a novel quasi-likelihood method for differential expression analysis of non-coding RNA'. BMC Genomics (2019) 20: 539.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data('hnsc.edata','tissue','cov')  

# DE analysis comparing two groups (normal vs tumor) for 100 genes
result=lncDIFF(edata=hnsc.edata[1:100,],group=tissue,covariate=cov) 

# Recommend at least 50 permutations if simulated.pvalue=TRUE




</code></pre>


</div>