<div class="container">

<table style="width: 100%;"><tr>
<td>make_referFreq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Make Frequencies File for Log.Dist, Euc.Dist, and hexamer score</h2>

<h3>Description</h3>

<p>This function is used to calculate the frequencies of lncRNAs and CDs.
The Frequencies file can be used to calculate Logarithm-Distance (<code>compute_LogDistance</code>),
Euclidean-Distance (<code>compute_EucDistance</code>), and hexamer score (<code>compute_hexamerScore</code>).
</p>
<p>NOTE: If users need to make frequencies file to build
new LncFinder classifier using function <code>extract_features</code>,
please refer to function <code>make_frequencies</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">make_referFreq(
  cds.seq,
  lncRNA.seq,
  k = 6,
  step = 1,
  alphabet = c("a", "c", "g", "t"),
  on.orf = TRUE,
  ignore.illegal = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cds.seq</code></td>
<td>
<p>Coding sequences (mRNA without UTRs). Can be a FASTA file loaded
by <code>seqinr-package</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lncRNA.seq</code></td>
<td>
<p>Long non-coding RNA sequences. Can be a FASTA file loaded by
<code>seqinr-package</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>An integer that indicates the sliding window size. (Default: <code>6</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>step</code></td>
<td>
<p>Integer defaulting to <code>1</code> for the window step.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphabet</code></td>
<td>
<p>A vector of single characters that specify the different character
of the sequence. (Default: <code>alphabet = c("a", "c", "g", "t")</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>on.orf</code></td>
<td>
<p>Logical. Incomplete CDs can lead to a false shift and a
inaccurate hexamer frequencies. When <code>on.orf = TRUE</code>, the frequencies
will be calculated on the longest ORF. This parameter is strongly recommended to
set as <code>TRUE</code> when mRNA is used as CDs. Only available when
<code>alphabet = c("a", "c", "g", "t")</code>. (Default: <code>TRUE</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore.illegal</code></td>
<td>
<p>Logical. If <code>TRUE</code>, the sequences with non-nucleotide
characters (nucleotide characters: "a", "c", "g", "t") will be ignored when
calculating the frequencies. Only available when <code>alphabet = c("a", "c", "g", "t")</code>.
(Default: <code>TRUE</code>)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is used to make frequencies file for the computation of
Logarithm-Distance (<code>compute_LogDistance</code>), Euclidean-Distance
(<code>compute_EucDistance</code>),
and hexamer score (<code>compute_hexamerScore</code>).
</p>
<p>In order to achieve high accuracy, mRNA should not be regarded as CDs and assigned
to parameter <code>cds.seq</code>. However, CDs of some species may be insufficient
for calculating frequencies. In that case, mRNAs can be regarded as CDs with parameter
<code>on.orf = TRUE</code>, and the frequencies will be calculated on ORF region.
If <code>on.orf = TRUE</code>, users can set <code>step = 3</code> to simulate the translation process.
</p>


<h3>Value</h3>

<p>Returns a list which consists the frequencies of protein-coding sequences
and non-coding sequences.
</p>


<h3>References</h3>

<p>Siyu Han, Yanchun Liang, Qin Ma, Yangyi Xu, Yu Zhang, Wei Du, Cankun Wang &amp; Ying Li.
LncFinder: an integrated platform for long non-coding RNA identification utilizing
sequence intrinsic composition, structural information, and physicochemical property.
<em>Briefings in Bioinformatics</em>, 2019, 20(6):2009-2027.
</p>


<h3>Author(s)</h3>

<p>HAN Siyu
</p>


<h3>See Also</h3>

<p><code>make_frequencies</code>,
<code>compute_LogDistance</code>,
<code>compute_EucDistance</code>,
<code>compute_hexamerScore</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
Seqs &lt;- seqinr::read.fasta(file =
"http://www.ncbi.nlm.nih.gov/WebSub/html/help/sample_files/nucleotide-sample.txt")

referFreq &lt;- make_referFreq(cds.seq = Seqs, lncRNA.seq = Seqs, k = 6, step = 1,
                            alphabet = c("a", "c", "g", "t"), on.orf = TRUE,
                            ignore.illegal = TRUE)

## End(Not run)

</code></pre>


</div>