<div class="container">

<table style="width: 100%;"><tr>
<td>find_orfs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find ORFs</h2>

<h3>Description</h3>

<p>This function can find all the ORFs in one sequence.
</p>


<h3>Usage</h3>

<pre><code class="language-R">find_orfs(OneSeq, reverse.strand = FALSE, max.only = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>OneSeq</code></td>
<td>
<p>Is one sequence. Can be a FASTA file read by package "seqinr"
<code>seqinr-package</code> or just a string.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reverse.strand</code></td>
<td>
<p>Logical. Whether find ORF on the reverse strand. Default: <code>FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.only</code></td>
<td>
<p>Logical. If <code>TRUE</code>, only the longest ORF will be returned.  Default: <code>TRUE</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function can extract ORFs of one sequence. It returns
ORF region, length and coverage of the longest ORF when <code>max.only = TRUE</code> or
ORF region, start position, end position, length and coverage of all the ORFs when
<code>max.only = FALSE</code>. Coverage is the the ratio
of the ORF to transcript length. If <code>reverse.strand = TRUE</code>, ORF will also be
found on reverse strand.
</p>


<h3>Value</h3>

<p>If <code>max.only = TRUE</code>, the function returns a list which consists the ORF region (<code>ORF.Max.Seq</code>),
length (<code>ORF.Max.Len</code>) and coverage (<code>ORF.Max.Cov</code>) of the longest ORF.
If <code>max.only = FALSE</code>, the function returns a dataframe which consists all the ORF sequences.
</p>


<h3>Author(s)</h3>

<p>HAN Siyu
</p>


<h3>Examples</h3>

<pre><code class="language-R">### For one sequence:
OneSeq &lt;- c("cccatgcccagctagtaagcttagcc")
orf.info_1 &lt;- find_orfs(OneSeq, reverse.strand = TRUE, max.only = FALSE)

### For a FASTA file contains several sequences:
## Not run: 
### Use "read.fasta" function of package "seqinr" to read a FASTA file:
Seqs &lt;- seqinr::read.fasta(file =
"http://www.ncbi.nlm.nih.gov/WebSub/html/help/sample_files/nucleotide-sample.txt")

## End(Not run)

### Or just try to use our data "demo_DNA.seq"
data(demo_DNA.seq)
Seqs &lt;- demo_DNA.seq

### Use apply function to find the longest ORF:
orf.info_2 &lt;- sapply(Seqs, find_orfs, reverse.strand = FALSE, max.only = FALSE)
</code></pre>


</div>