<div class="container">

<table style="width: 100%;"><tr>
<td>findSigGenes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>

Find genes significantly differentially expressed between two conditions.
</h2>

<h3>Description</h3>


<p>For a given expression profile of two conditions, find the genes differencially expressed using T-test, fold change or SAM algorithm.
</p>


<h3>Usage</h3>

<pre><code class="language-R">findSigGenes(Expr, Label, Method = "tTest", Directed = TRUE, 
	FdrCut = 0.01, FDCut = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Expr</code></td>
<td>


<p>A data frame, the expression profile to find differentially expressed genes, the rownames should be the ID of genes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Label</code></td>
<td>


<p>A vector of 0/1s, indicating the class of samples in the expression profile, 0 represents case, 1 represents control.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Method</code></td>
<td>


<p>A string, specifying the method to calculate the differentially expressed genes, should be one of the "tTest"or"foldChange".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Directed</code></td>
<td>


<p>Logical, if the the up or down regulated set should be distinguished.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FdrCut</code></td>
<td>


<p>Numeric, the fdr cutoff for T test, can be ignored if not using t-test.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FDCut</code></td>
<td>


<p>Numeric, the cutoff for fold change, can be ignored if not using fold change.
</p>
</td>
</tr>
</table>
<h3>Details</h3>


<p>For a given expression profile of two conditions, lncPath package provide two method to find differentially expressed genes: t-text and fold change. The row of the expression profile should be gene IDs and the column of the expression profile should be names of samples. Samples should be under two conditions and the label should be given as 0 and 1. For t-test, fold change and SAM, different threshold can be set for significant differentially expressed genes.  
</p>


<h3>Value</h3>






<p>A vector of strings, the IDs of differentially expressed genes.
</p>


<h3>Author(s)</h3>


<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;
</p>


<h3>References</h3>


<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class="language-R">##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.
Profile &lt;- getExampleData("Profile")
Labels &lt;- getExampleData("Labels")

SigGenes &lt;- findSigGenes(Profile, Labels)
head(SigGenes)

</code></pre>


</div>