<div class="container">

<table style="width: 100%;"><tr>
<td>quick_peak</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Fast peak finder in GWAS data</h2>

<h3>Description</h3>

<p>Simple but fast function for finding peaks in genome-wide association study
(GWAS) data based on setting a minimum distance between peaks.
</p>


<h3>Usage</h3>

<pre><code class="language-R">quick_peak(
  data,
  npeaks = NA,
  p_cutoff = 5e-08,
  span = 1e+06,
  min_points = 2,
  chrom = NULL,
  pos = NULL,
  p = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>GWAS dataset (data.frame or data.table)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>npeaks</code></td>
<td>
<p>Number of peaks to find. If set to <code>NA</code>, algorithm finds all
distinct peaks separated from one another by region size specified by
<code>span</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_cutoff</code></td>
<td>
<p>Specifies cut-off for p-value significance above which
p-values are ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>span</code></td>
<td>
<p>Minimum genomic distance between peaks (default 1 Mb)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_points</code></td>
<td>
<p>Minimum number of p-value significant points which must lie
within the span of a peak. This removes peaks with single or only a few low
p-value SNPs. To disable set <code>min_points</code> to 1 or less.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom</code></td>
<td>
<p>Determines which column in <code>data</code> contains chromosome
information. If <code>NULL</code> tries to autodetect the column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>Determines which column in <code>data</code> contains position information.
If <code>NULL</code> tries to autodetect the column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>Determines which column in <code>data</code> contains SNP p-values. If <code>NULL</code>
tries to autodetect the column.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is designed for speed. SNP p-values are filtered to only those
which are significant as specified by <code>p_cutoff</code>. Each peak is identified as
the SNP with the lowest p-value and then SNPs in proximity to each peak
within the distance specified by <code>span</code> are removed. Regions such as the HLA
whose peaks may well be broader than <code>span</code> may produce multiple entries.
</p>


<h3>Value</h3>

<p>Vector of row indices
</p>


</div>