<div class="container">

<table style="width: 100%;"><tr>
<td>locus_plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Locus plot</h2>

<h3>Description</h3>

<p>Genomic locus plot similar to locuszoom.
</p>


<h3>Usage</h3>

<pre><code class="language-R">locus_plot(
  loc,
  filter_gene_name = NULL,
  filter_gene_biotype = NULL,
  xlab = NULL,
  cex = 1,
  cex.axis = 0.9,
  cex.lab = 1,
  cex.text = 0.7,
  use_layout = TRUE,
  heights = c(3, 2),
  showExons = TRUE,
  maxrows = 7,
  xticks = "bottom",
  border = FALSE,
  gene_col = ifelse(showExons, "blue4", "skyblue"),
  exon_col = "blue4",
  exon_border = "blue4",
  text_pos = "top",
  highlight = NULL,
  highlight_col = "red",
  blanks = "fill",
  recomb_col = "blue",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <code>locus()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_gene_name</code></td>
<td>
<p>Vector of gene names to display.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_gene_biotype</code></td>
<td>
<p>Vector of gene biotypes to be filtered. Use
<code>ensembldb::listGenebiotypes()</code> to display possible biotypes. For example,
<code>ensembldb::listGenebiotypes(EnsDb.Hsapiens.v75)</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>x axis title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>Specifies size for points.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.lab</code></td>
<td>
<p>Specifies font size for axis titles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.text</code></td>
<td>
<p>Font size for gene text.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_layout</code></td>
<td>
<p>Logical whether <code>graphics::layout</code> is called. Default
<code>TRUE</code> is for a standard single plot. Set to <code>FALSE</code> if a more complex
layout with multiple plots is required e.g. using <code>multi_layout()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>heights</code></td>
<td>
<p>Ratio of top to bottom plot. See layout.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showExons</code></td>
<td>
<p>Logical whether to show exons or simply show whole gene as a
rectangle</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxrows</code></td>
<td>
<p>Specifies maximum number of rows to display in gene
annotation panel.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xticks</code></td>
<td>
<p>Character value of either 'top' or 'bottom' specifying
whether x axis ticks and numbers are plotted on top or bottom plot window.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>border</code></td>
<td>
<p>Logical whether a bounding box is plotted around upper and
lower plots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_col</code></td>
<td>
<p>Colour for gene lines.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exon_col</code></td>
<td>
<p>Fill colour for exons.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exon_border</code></td>
<td>
<p>Border line colour outlining exons (or genes if
<code>showExons</code> is <code>FALSE</code>). Set to <code>NA</code> for no border.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>text_pos</code></td>
<td>
<p>Character value of either 'top' or 'left' specifying
placement of gene name labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight</code></td>
<td>
<p>Vector of genes to highlight.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight_col</code></td>
<td>
<p>Single colour or vector of colours for highlighted
genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blanks</code></td>
<td>
<p>Controls handling of genes with blank names: <code>"fill"</code> replaces
blank gene symbols with ensembl gene ids. <code>"hide"</code> hides genes which are
missing gene symbols.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>recomb_col</code></td>
<td>
<p>Colour for recombination rate line if recombination rate
data is present. Set to <code>NA</code> to hide the line. See <code>link_recomb()</code> to add
recombination rate data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other arguments passed to <code>scatter_plot()</code> and <code>plot()</code> to control
the scatter plot, e.g. <code>ylab</code>, <code>main</code>, etc.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This is an R version of locuszoom for generating publication ready Manhattan
plots of gene loci. It references Ensembl databases for annotating genes
and exons. Use <code>locus()</code> first to generate an object of class 'locus' for
plotting. LDlink web server can be queried using function <code>link_LD()</code> to
retrieve linkage disequilibrium (LD) information on the index SNP.
</p>
<p>Arguments to control plotting of the gene tracks are passed onto
<code>genetracks()</code> and for the scatter plot are passed via <code>...</code> to
<code>scatter_plot()</code>. See the documentation for each of these functions for
details.
</p>


<h3>Value</h3>

<p>No return value.
</p>


<h3>See Also</h3>

<p><code>locus()</code> <code>scatter_plot()</code> <code>genetracks()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'UBE2L3', flank = 1e5,
             ens_db = "EnsDb.Hsapiens.v75")
locus_plot(loc)

## Use embedded LD information in column `r2`
loc2 &lt;- locus(SLE_gwas_sub, gene = 'IRF5', flank = c(7e4, 2e5), LD = "r2",
              ens_db = "EnsDb.Hsapiens.v75")
## Add label for index SNP
locus_plot(loc2, labels = "index")
}
</code></pre>


</div>