<div class="container">

<table style="width: 100%;"><tr>
<td>genetrack_ly</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gene tracks using 'plotly'</h2>

<h3>Description</h3>

<p>Plot gene annotation tracks from <code>ensembldb</code> data using <code>plotly</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">genetrack_ly(
  locus,
  filter_gene_name = NULL,
  filter_gene_biotype = NULL,
  cex.text = 0.7,
  gene_col = ifelse(showExons, "blue4", "skyblue"),
  exon_col = "blue4",
  exon_border = "blue4",
  showExons = TRUE,
  maxrows = 8,
  width = 600,
  xlab = NULL,
  blanks = c("fill", "hide", "show"),
  height = NULL,
  plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>locus</code></td>
<td>
<p>Object of class 'locus' generated by <code>locus()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_gene_name</code></td>
<td>
<p>Vector of gene names to display.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_gene_biotype</code></td>
<td>
<p>Vector of gene biotypes to be filtered. Use
<code>ensembldb::listGenebiotypes()</code> to display possible biotypes. For example,
<code>ensembldb::listGenebiotypes(EnsDb.Hsapiens.v75)</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.text</code></td>
<td>
<p>Font size for gene text.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_col</code></td>
<td>
<p>Colour for gene lines.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exon_col</code></td>
<td>
<p>Fill colour for exons.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exon_border</code></td>
<td>
<p>Border line colour outlining exons (or genes if
<code>showExons</code> is <code>FALSE</code>). Set to <code>NA</code> for no border.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showExons</code></td>
<td>
<p>Logical whether to show exons or simply show whole gene as a
rectangle. If <code>showExons = FALSE</code> colours are specified by <code>exon_border</code>
for rectangle border and <code>gene_col</code> for the fill colour.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxrows</code></td>
<td>
<p>Specifies maximum number of rows to display in gene
annotation panel.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>width</code></td>
<td>
<p>Width of plotly plot in pixels which is purely used to prevent
overlapping text for gene names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>Title for x axis. Defaults to chromosome <code>seqname</code> specified
in <code>locus</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blanks</code></td>
<td>
<p>Controls handling of genes with blank names: <code>"fill"</code> replaces
blank gene symbols with ensembl gene ids. <code>"hide"</code> completely hides genes
which are missing gene symbols. <code>"show"</code> shows gene lines but no label
(hovertext is still available).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>height</code></td>
<td>
<p>Height in pixels (optional, defaults to automatic sizing).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>Logical whether to produce plotly object or return plot
coordinates.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function can used to plot gene annotation tracks on their own.
</p>


<h3>Value</h3>

<p>Either a 'plotly' plotting object showing gene tracks, or if
<code>plot = FALSE</code> a list containing <code>TX</code>, a dataframe of coordinates for
gene transcripts, and <code>EX</code>, a dataframe of coordinates for exons.
</p>


<h3>Examples</h3>

<pre><code class="language-R">if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'UBE2L3', flank = 1e5,
             ens_db = "EnsDb.Hsapiens.v75")
genetrack_ly(loc)
}
</code></pre>


</div>