<div class="container">

<table style="width: 100%;"><tr>
<td>locus</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create locus object for plotting</h2>

<h3>Description</h3>

<p>Creates object of class 'locus' for genomic locus plot similar to
<code>locuszoom</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">locus(
  gene = NULL,
  data = NULL,
  xrange = NULL,
  seqname = NULL,
  flank = NULL,
  fix_window = NULL,
  ens_db,
  chrom = NULL,
  pos = NULL,
  p = NULL,
  yvar = NULL,
  labs = NULL,
  index_snp = NULL,
  LD = NULL,
  std_filter = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gene</code></td>
<td>
<p>Optional character value specifying which gene to view. Either
<code>gene</code>, or <code>xrange</code> plus <code>seqname</code>, or <code>index_snp</code> must be specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Dataset (data.frame or data.table) to use for plot. If
unspecified or <code>NULL</code>, gene track information alone is returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xrange</code></td>
<td>
<p>Optional vector of genomic position range for the x axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqname</code></td>
<td>
<p>Optional, specifies which chromosome to plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flank</code></td>
<td>
<p>Single value or vector with 2 values for how much flanking
region left and right of the gene to show. Defaults to 100kb.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fix_window</code></td>
<td>
<p>Optional alternative to <code>flank</code>, which allows users to
specify a fixed genomic window centred on the specified gene. Both <code>flank</code>
and <code>fix_window</code> cannot be specified simultaneously.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ens_db</code></td>
<td>
<p>Either a character string which specifies which Ensembl
database package (version 86 and earlier for Homo sapiens) to query for
gene and exon positions (see <code>ensembldb</code> Bioconductor package). Or an
<code>ensembldb</code> object which can be obtained from the AnnotationHub database.
See the vignette and the <code>AnnotationHub</code> Bioconductor package for how to
create this object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom</code></td>
<td>
<p>Determines which column in <code>data</code> contains chromosome
information. If <code>NULL</code> tries to autodetect the column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>Determines which column in <code>data</code> contains position information.
If <code>NULL</code> tries to autodetect the column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>Determines which column in <code>data</code> contains SNP p-values.
If <code>NULL</code> tries to autodetect the column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yvar</code></td>
<td>
<p>Specifies column in <code>data</code> for plotting on the y axis as an
alternative to specifying p-values. Both <code>p</code> and <code>yvar</code> cannot be specified
simultaneously.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labs</code></td>
<td>
<p>Determines which column in <code>data</code> contains SNP rs IDs.
If <code>NULL</code> tries to autodetect the column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index_snp</code></td>
<td>
<p>Specifies the index SNP. If not specified, the SNP with the
lowest P value is selected. Can be used to specify locus region instead of
specifying <code>gene</code>, or <code>seqname</code> and <code>xrange</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LD</code></td>
<td>
<p>Optional character value to specify which column in <code>data</code> contains
LD information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>std_filter</code></td>
<td>
<p>Logical, whether standard filters on chromosomes 1-22, X &amp;
Y, and filtering of genes to only those whose transcript ids start with
"ENS" are applied. For users with novel genome assemblies, this probably
needs to be set to <code>FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This is an R version of <code>locuszoom</code> (http://locuszoom.org) for generating
publication ready Manhattan plots of gene loci. It references Ensembl
databases using the <code>ensembldb</code> Bioconductor package framework for annotating
genes and exons in the locus.
</p>


<h3>Value</h3>

<p>Returns a list object of class 'locus' ready for plotting,
containing:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>seqname</code></td>
<td>
<p>chromosome value</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xrange</code></td>
<td>
<p>vector of genomic position range</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene</code></td>
<td>
<p>gene name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ens_db</code></td>
<td>
<p>Ensembl or AnnotationHub database</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ens_version</code></td>
<td>
<p>Ensembl database version</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organism</code></td>
<td>
<p>Ensembl database organism</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>Ensembl data genome build</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom</code></td>
<td>
<p>column name in <code>data</code> containing chromosome information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>column name in <code>data</code> containing position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>column name in <code>data</code> containing p-value</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yvar</code></td>
<td>
<p>column name in <code>data</code> to be plotted on y axis as alternative to
<code>p</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labs</code></td>
<td>
<p>column name in <code>data</code> containing SNP IDs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index_snp</code></td>
<td>
<p>id of the most significant SNP</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>the subset of GWAS data to be plotted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TX</code></td>
<td>
<p>dataframe of transcript annotations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EX</code></td>
<td>
<p><code>GRanges</code> object of exon annotations</p>
</td>
</tr>
</table>
<p>If <code>data</code> is <code>NULL</code> when <code>locus()</code> is called then gene track information
alone is returned.
</p>


<h3>See Also</h3>

<p><code>locus_plot()</code> <code>locus_ggplot()</code> <code>locus_plotly()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Bioconductor package EnsDb.Hsapiens.v75 is needed for these examples
if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'UBE2L3', flank = 1e5,
             ens_db = "EnsDb.Hsapiens.v75")
summary(loc)
locus_plot(loc)
loc2 &lt;- locus(SLE_gwas_sub, gene = 'STAT4', flank = 1e5,
              ens_db = "EnsDb.Hsapiens.v75")
locus_plot(loc2)
}
</code></pre>


</div>