<div class="container">

<table style="width: 100%;"><tr>
<td>overlay_plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot overlaying eQTL and GWAS data</h2>

<h3>Description</h3>

<p>Experimental plotting function for overlaying eQTL data from GTEx on top of
GWAS results. y axis shows the -log10 p-value for the GWAS result.
Significant eQTL for the specified gene are overlaid using colours and
symbols.
</p>


<h3>Usage</h3>

<pre><code class="language-R">overlay_plot(
  loc,
  base_col = "black",
  alpha = 0.5,
  scheme = "RdYlBu",
  tissue = "Whole Blood",
  eqtl_gene = loc$gene,
  legend_pos = "topright",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <code>locus()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base_col</code></td>
<td>
<p>Colour of points for SNPs which do not have eQTLs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>Alpha opacity for non-eQTL points</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scheme</code></td>
<td>
<p>Character string specifying palette for effect size showing
up/downregulation eQTL using grDevices::hcl.colors. Alternatively a
vector of 6 colours.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tissue</code></td>
<td>
<p>GTex tissue in which eQTL has been measured</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eqtl_gene</code></td>
<td>
<p>Gene showing eQTL effect</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_pos</code></td>
<td>
<p>Character value specifying legend position. See <code>legend()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other arguments passed to <code>locus_plot()</code> for the locus plot.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>No return value. Produces a plot using base graphics.
</p>


</div>