<div class="container">

<table style="width: 100%;"><tr>
<td>scatter_plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Locus scatter plot</h2>

<h3>Description</h3>

<p>Produces a base graphics scatter plot from a 'locus' class object. This
function is called by <code>locus_plot()</code> to generate the scatter plot portion.
Can be used manually with <code>set_layers()</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">scatter_plot(
  loc,
  index_snp = loc$index_snp,
  pcutoff = 5e-08,
  scheme = c("grey", "dodgerblue", "red"),
  cex = 1,
  cex.axis = 0.9,
  cex.lab = 1,
  xlab = NULL,
  ylab = NULL,
  yzero = (loc$yvar == "logP"),
  xticks = TRUE,
  border = FALSE,
  showLD = TRUE,
  LD_scheme = c("grey", "royalblue", "cyan2", "green3", "orange", "red", "purple"),
  recomb_col = "blue",
  legend_pos = "topleft",
  labels = NULL,
  label_x = 4,
  label_y = 4,
  eqtl_gene = NULL,
  beta = NULL,
  add = FALSE,
  align = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index_snp</code></td>
<td>
<p>Specifies index SNP or a vector of SNPs to be shown in a
different colour and symbol. Defaults to the SNP with the lowest p-value.
Set to <code>NULL</code> to not show this.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pcutoff</code></td>
<td>
<p>Cut-off for p value significance. Defaults to p = 5e-08. Set
to <code>NULL</code> to disable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scheme</code></td>
<td>
<p>Vector of 3 colours if LD is not shown: 1st = normal points,
2nd = colour for significant points, 3rd = index SNP(s).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>Specifies size for points.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.lab</code></td>
<td>
<p>Specifies font size for axis titles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>x axis title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>y axis title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yzero</code></td>
<td>
<p>Logical whether to force y axis limit to include y=0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xticks</code></td>
<td>
<p>Logical whether x axis numbers and axis title are plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>border</code></td>
<td>
<p>Logical whether a bounding box is plotted around upper and
lower plots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showLD</code></td>
<td>
<p>Logical whether to show LD with colours</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LD_scheme</code></td>
<td>
<p>Vector of colours for plotting LD. The first colour is for
SNPs which lack LD information. The next 5 colours are for r2 or D' LD
results ranging from 0 to 1 in intervals of 0.2. The final colour is for
the index SNP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>recomb_col</code></td>
<td>
<p>Colour for recombination rate line if recombination rate
data is present. Set to <code>NA</code> to hide the line. See <code>link_recomb()</code> to add
recombination rate data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_pos</code></td>
<td>
<p>Position of legend. See <code>legend()</code>. Set to <code>NULL</code> to hide
legend.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labels</code></td>
<td>
<p>Character vector of SNP or genomic feature IDs to label. The
value "index" selects the highest point or index SNP as defined when
<code>locus()</code> is called. Set to <code>NULL</code> to remove all labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_x</code></td>
<td>
<p>Value or vector for position of label as percentage of x axis
scale.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_y</code></td>
<td>
<p>Value or vector for position of label as percentage of y axis
scale.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eqtl_gene</code></td>
<td>
<p>Column name in <code>loc$data</code> for colouring eQTL genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta</code></td>
<td>
<p>Optional column name for beta coefficient to display upward
triangles for positive beta and downward triangles for negative beta
(significant SNPs only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add</code></td>
<td>
<p>Logical whether to add points to an existing plot or generate a
new plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>align</code></td>
<td>
<p>Logical whether to set <code>par()</code> to align the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other arguments passed to <code>plot()</code> to control the scatter plot
e.g. <code>main</code>, <code>ylim</code> etc.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Advanced users familiar with base graphics can customise every single point
on the scatter plot, by adding columns named <code>bg</code>, <code>col</code>, <code>pch</code> or <code>cex</code>
directly to the dataframe stored in <code style="white-space: pre;">⁠$data⁠</code> element of the 'locus' object.
Setting these will overrule any default settings. These columns refer to
their respective base graphics arguments, see <code>graphics::points()</code>.
</p>


<h3>Value</h3>

<p>No return value. Produces a scatter plot using base graphics.
</p>


<h3>See Also</h3>

<p><code>locus()</code> <code>set_layers()</code>
</p>


</div>