<div class="container">

<table style="width: 100%;"><tr>
<td>gg_scatter</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Locus scatter plot using ggplot2</h2>

<h3>Description</h3>

<p>Produces a scatter plot from a 'locus' class object (without gene tracks).
</p>


<h3>Usage</h3>

<pre><code class="language-R">gg_scatter(
  loc,
  index_snp = loc$index_snp,
  pcutoff = 5e-08,
  scheme = c("grey", "dodgerblue", "red"),
  size = 2,
  cex.axis = 1,
  cex.lab = 1,
  xlab = NULL,
  ylab = NULL,
  yzero = (loc$yvar == "logP"),
  xticks = TRUE,
  border = FALSE,
  showLD = TRUE,
  LD_scheme = c("grey", "royalblue", "cyan2", "green3", "orange", "red", "purple"),
  recomb_col = "blue",
  legend_pos = "topleft",
  labels = NULL,
  eqtl_gene = NULL,
  beta = NULL,
  shape = NULL,
  shape_values = c(21, 24, 25),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index_snp</code></td>
<td>
<p>Specifies index SNP to be shown in a different colour and
symbol. Defaults to the SNP with the lowest p-value. Set to <code>NULL</code> to not
show this.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pcutoff</code></td>
<td>
<p>Cut-off for p value significance. Defaults to p = 5e-08. Set
to <code>NULL</code> to disable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scheme</code></td>
<td>
<p>Vector of 3 colours if LD is not shown: 1st = normal points,
2nd = colour for significant points, 3rd = index SNP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p>Specifies size for points.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.lab</code></td>
<td>
<p>Specifies font size for axis titles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>x axis title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>y axis title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yzero</code></td>
<td>
<p>Logical whether to force y axis limit to include y=0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xticks</code></td>
<td>
<p>Logical whether x axis numbers and axis title are plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>border</code></td>
<td>
<p>Logical whether a bounding box is plotted around the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showLD</code></td>
<td>
<p>Logical whether to show LD with colours</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LD_scheme</code></td>
<td>
<p>Vector of colours for plotting LD. The first colour is for SNPs
which lack LD information. The next 5 colours are for r2 or D' LD results
ranging from 0 to 1 in intervals of 0.2. The final colour is for the index
SNP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>recomb_col</code></td>
<td>
<p>Colour for recombination rate line if recombination rate
data is present. Set to NA to hide the line. See <code>link_recomb()</code> to add
recombination rate data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_pos</code></td>
<td>
<p>Position of legend. Set to <code>NULL</code> to hide legend.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labels</code></td>
<td>
<p>Character vector of SNP or genomic feature IDs to label. The
value "index" selects the highest point or index SNP as defined when
<code>locus()</code> is called. Set to <code>NULL</code> to remove all labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eqtl_gene</code></td>
<td>
<p>Optional column name in <code>loc$data</code> for colouring eQTL genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta</code></td>
<td>
<p>Optional column name for beta coefficient to display upward
triangles for positive beta and downward triangles for negative beta
(significant SNPs only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shape</code></td>
<td>
<p>Optional column name in <code>loc$data</code> for controlling shapes.
<code>beta</code> and <code>shape</code> cannot both be set. This column is expected to be a factor.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shape_values</code></td>
<td>
<p>Vector of shape values which match levels of the column
specified by <code>shape</code>. This vector is passed to
<code>ggplot2::scale_shape_manual()</code> as the argument <code>values</code>. See <code>points()</code>
for a list of shapes and the numbers they map to.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Optional arguments passed to <code>geom_text_repel()</code> to configure
label drawing.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If recombination rate data is included in the locus object following a call
to <code>link_recomb()</code>, this is plotted as an additional line with a secondary y
axis. In the base graphics version the line is placed under the scatter
points, but this is not possible with ggplot2 as the secondary y axis data
must be plotted on top of the primary scatter point data.
</p>


<h3>Value</h3>

<p>Returns a ggplot2 plot.
</p>


<h3>See Also</h3>

<p><code>locus()</code> <code>gg_addgenes()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'IRF5', flank = c(7e4, 2e5), LD = "r2",
             ens_db = "EnsDb.Hsapiens.v75")
gg_scatter(loc)
}
</code></pre>


</div>