<div class="container">

<table style="width: 100%;"><tr>
<td>link_LD</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Obtain LD at a locus from LDlink</h2>

<h3>Description</h3>

<p>Adds LD information to a 'locus' class object. It queries LDlink
(https://ldlink.nci.nih.gov/) via the <code>LDlinkR</code> package to retrieve linkage
disequilibrium (LD) information on a reference SNP.
</p>


<h3>Usage</h3>

<pre><code class="language-R">link_LD(
  loc,
  pop = "CEU",
  r2d = "r2",
  token = "",
  method = c("proxy", "matrix"),
  genome_build = loc$genome,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>loc</code></td>
<td>
<p>Object of class 'locus' generated by <code>locus()</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>A 1000 Genomes Project population, (e.g. YRI or CEU), multiple
allowed, default = "CEU". Passed to <code>LDlinkR::LDmatrix()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r2d</code></td>
<td>
<p>Either "r2" for LD r^2 or "d" for LD D', default = "r2". Passed
to <code>LDlinkR::LDmatrix()</code> or <code>LDproxy()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>token</code></td>
<td>
<p>Personal access token for accessing 1000 Genomes LD data via
LDlink API. See <code>LDlinkR</code> package documentation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Either <code>"proxy"</code> or <code>"matrix"</code>. Controls whether to use
<code>LDproxy()</code> or <code>LDmatrix()</code> to obtain LD data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_build</code></td>
<td>
<p>Choose between one of the three options: 'grch37' for
genome build GRCh37 (hg19), 'grch38' for GRCh38 (hg38), or
'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets. Default is GRCh37 (hg19).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Optional arguments which are passed on to <code>LDlinkR::LDmatrix()</code> or
<code>LDlinkR::LDproxy()</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The argument <code>method</code> controls which LDlinkR function is used to retrieve LD
data. <code>LDmatrix()</code> is slower but usually more complete for small queries
(&lt;1000 SNPs). However, it has a limit of 1000 SNPs which can be queried.
<code>LDproxy()</code> is faster but data on some SNPs may be absent.
</p>
<p>Note, SNPs have to be correctly formatted as required by LDlinkR, either as
rsID (works with either method) or chromosome coordinate e.g. "chr7:24966446"
(works with LDproxy only). Default genome build is <code>grch37</code>, see <code>LDproxy()</code>
or <code>LDmatrix()</code>.
</p>


<h3>Value</h3>

<p>Returns a list object of class 'locus'. LD information is added as a
column <code>ld</code> in list element <code>data</code>.
</p>


<h3>See Also</h3>

<p><code>locus()</code>
</p>


</div>