<div class="container">

<table style="width: 100%;"><tr>
<td>locus_plotly</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Locus plotly</h2>

<h3>Description</h3>

<p>Genomic locus plot similar to locuszoom, using plotly.
</p>


<h3>Usage</h3>

<pre><code class="language-R">locus_plotly(
  loc,
  heights = c(0.6, 0.4),
  filter_gene_name = NULL,
  filter_gene_biotype = NULL,
  cex.text = 0.7,
  gene_col = ifelse(showExons, "blue4", "skyblue"),
  exon_col = "blue4",
  exon_border = "blue4",
  showExons = TRUE,
  maxrows = 8,
  width = 600,
  xlab = NULL,
  blanks = "show",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <code>locus()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>heights</code></td>
<td>
<p>Vector controlling relative height of each panel on 0-1 scale.
Alternatively a vector of length 2 of height in pixels passed to
<code>scatter_plotly()</code> and <code>genetrack_ly()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_gene_name</code></td>
<td>
<p>Vector of gene names to display.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_gene_biotype</code></td>
<td>
<p>Vector of gene biotypes to be filtered. Use
<code>ensembldb::listGenebiotypes()</code> to display possible biotypes. For example,
<code>ensembldb::listGenebiotypes(EnsDb.Hsapiens.v75)</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.text</code></td>
<td>
<p>Font size for gene text.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_col</code></td>
<td>
<p>Colour for gene lines.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exon_col</code></td>
<td>
<p>Fill colour for exons.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exon_border</code></td>
<td>
<p>Border line colour outlining exons (or genes if
<code>showExons</code> is <code>FALSE</code>). Set to <code>NA</code> for no border.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showExons</code></td>
<td>
<p>Logical whether to show exons or simply show whole gene as a
rectangle. If <code>showExons = FALSE</code> colours are specified by <code>exon_border</code>
for rectangle border and <code>gene_col</code> for the fill colour.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxrows</code></td>
<td>
<p>Specifies maximum number of rows to display in gene
annotation panel.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>width</code></td>
<td>
<p>Width of plotly plot in pixels which is purely used to prevent
overlapping text for gene names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>Title for x axis. Defaults to chromosome <code>seqname</code> specified
in <code>locus</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blanks</code></td>
<td>
<p>Controls handling of genes with blank names: <code>"fill"</code> replaces
blank gene symbols with ensembl gene ids. <code>"hide"</code> completely hides genes
which are missing gene symbols. <code>"show"</code> shows gene lines but no label
(hovertext is still available).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Optional arguments passed to <code>scatter_plotly()</code> to control the
scatter plot.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This is an R/plotly version of locuszoom for exploring regional Manhattan
plots of gene loci. Use <code>locus()</code> first to generate an object of class
'locus' for plotting. This references a selected Ensembl database for
annotating genes and exons. Hover over the points or gene tracks to reveal
more information.
</p>


<h3>Value</h3>

<p>A 'plotly' plotting object showing a scatter plot above gene tracks.
</p>


<h3>See Also</h3>

<p><code>locus()</code> <code>genetrack_ly()</code> <code>scatter_plotly()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = "IRF5", flank = c(7e4, 2e5), LD = "r2",
             ens_db = "EnsDb.Hsapiens.v75")
locus_plotly(loc)
}
</code></pre>


</div>