<div class="container">

<table style="width: 100%;"><tr>
<td>gg_genetracks</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot gene tracks</h2>

<h3>Description</h3>

<p>Plot gene annotation tracks from <code>ensembldb</code> data using ggplot2 and grid.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gg_genetracks(
  loc,
  filter_gene_name = NULL,
  filter_gene_biotype = NULL,
  border = FALSE,
  cex.axis = 1,
  cex.lab = 1,
  cex.text = 0.7,
  gene_col = ifelse(showExons, "blue4", "skyblue"),
  exon_col = "blue4",
  exon_border = "blue4",
  showExons = TRUE,
  maxrows = NULL,
  text_pos = "top",
  xticks = TRUE,
  xlab = NULL,
  highlight = NULL,
  highlight_col = "red",
  blanks = c("fill", "hide")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>loc</code></td>
<td>
<p>Object of class 'locus' generated by <code>locus()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_gene_name</code></td>
<td>
<p>Vector of gene names to display.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_gene_biotype</code></td>
<td>
<p>Vector of gene biotypes to be filtered. Use
<code>ensembldb::listGenebiotypes()</code> to display possible biotypes. For example,
<code>ensembldb::listGenebiotypes(EnsDb.Hsapiens.v75)</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>border</code></td>
<td>
<p>Logical whether a bounding box is plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.lab</code></td>
<td>
<p>Specifies font size for axis titles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.text</code></td>
<td>
<p>Font size for gene text.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_col</code></td>
<td>
<p>Colour for gene lines.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exon_col</code></td>
<td>
<p>Fill colour for exons.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exon_border</code></td>
<td>
<p>Border line colour outlining exons (or genes if
<code>showExons</code> is <code>FALSE</code>). Set to <code>NA</code> for no border.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showExons</code></td>
<td>
<p>Logical whether to show exons or simply show whole gene as a
rectangle. If <code>showExons = FALSE</code> colours are specified by <code>exon_border</code>
for rectangle border and <code>gene_col</code> for the fill colour.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxrows</code></td>
<td>
<p>Specifies maximum number of rows to display in gene annotation
panel.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>text_pos</code></td>
<td>
<p>Character value of either 'top' or 'left' specifying
placement of gene name labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xticks</code></td>
<td>
<p>Logical whether x axis ticks and numbers are plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>Title for x axis. Defaults to chromosome <code>seqname</code> specified in
<code>locus</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight</code></td>
<td>
<p>Vector of genes to highlight.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight_col</code></td>
<td>
<p>Single colour or vector of colours for highlighted
genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blanks</code></td>
<td>
<p>Controls handling of genes with blank names: <code>"fill"</code> replaces
blank gene symbols with ensembl gene ids. <code>"hide"</code> hides genes which are
missing gene symbols.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is called by <code>locus_ggplot()</code>, and in turn it calls
<code>genetracks_grob()</code>. It can be used to plot the gene annotation tracks on
their own as a ggplot2 object.
</p>
<p><code>gene_col</code>, <code>exon_col</code> and <code>exon_border</code> set colours for all genes, while
<code>highlight</code> and <code>highlight_col</code> can optionally be used together to highlight
specific genes of interest. For full control over every single gene, users
can add columns <code>gene_col</code>, <code>exon_col</code> and <code>exon_border</code> to the <code>TX</code> object
within the 'locus' object. Columns added to <code>TX</code> override their equivalent
arguments.
</p>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>See Also</h3>

<p><code>locus_ggplot()</code> <code>genetracks_grob()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'IRF5', flank = c(7e4, 2e5), LD = "r2",
             ens_db = "EnsDb.Hsapiens.v75")
gg_genetracks(loc)
}
</code></pre>


</div>