<div class="container">

<table style="width: 100%;"><tr>
<td>lolliplot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Lolliplots</h2>

<h3>Description</h3>

<p>Plot variants and somatic mutations.
Jianhong Ou and Lihua Julie Zhu (2019)&lt;doi:10.1038/s41592-019-0430-y&gt;
</p>


<h3>Usage</h3>

<pre><code class="language-R">lolliplot(
  SNP.gr,
  features = NULL,
  ranges = NULL,
  type = "circle",
  newpage = TRUE,
  ylab = TRUE,
  ylab.gp = gpar(col = "black"),
  yaxis = TRUE,
  yaxis.gp = gpar(col = "black"),
  xaxis = TRUE,
  xaxis.gp = gpar(col = "black"),
  legend = NULL,
  cex = 1,
  dashline.col = "gray80",
  jitter = c("node", "label"),
  rescale = FALSE,
  label_on_feature = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SNP.gr</code></td>
<td>
<p>A object of GRanges,
GRangesList
or a list of GRanges.
All the width of GRanges must be 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>
<p>A object of GRanges,
GRangesList
or a list of GRanges.
The metadata 'featureLayerID' are used for drawing features in different layers.
See details in vignette.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ranges</code></td>
<td>
<p>A object of GRanges or
GRangesList.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>character. Could be circle, pie, pin, pie.stack or flag.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>newpage</code></td>
<td>
<p>Plot in the new page or not.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>Plot ylab or not. If it is a character vector,
the vector will be used as ylab.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab.gp, xaxis.gp, yaxis.gp</code></td>
<td>
<p>An object of class gpar for ylab, xaxis or yaxis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yaxis</code></td>
<td>
<p>Plot yaxis or not.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xaxis</code></td>
<td>
<p>Plot xaxis or not. If it is a numeric vector with length greater than 1,
the vector will be used as the points at which tick-marks are to be drawn.
And the names of the vector will be used to as labels to be placed at the tick
points if it has names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend</code></td>
<td>
<p>If it is a list with named color vectors, a legend will be added.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>cex will control the size of circle.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dashline.col</code></td>
<td>
<p>color for the dashed line.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>jitter</code></td>
<td>
<p>jitter the position of nodes or labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rescale</code></td>
<td>
<p>logical(1), character(1), numeric vector,
or a dataframe with rescale from and to.
Rescalse the x-axis or not.
if dataframe is used, colnames must be from.start, from.end,
to.start, to.end. And the from scale must cover the whole plot region.
The rescale parameter can be set as "exon" or "intron" to
emphasize "exon" or "intron" region. The "exon" or "intron"
can be followed with an integer e.g. "exon_80", or "intron_99".
The integer indicates the total percentage of "exon" or "intron" region.
Here "exon" indicates all regions in features.
And "intron" indicates all flank regions of the features.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_on_feature</code></td>
<td>
<p>Labels of the feature directly on them.
Default FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>not used.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In SNP.gr and features, metadata of the GRanges object will be used to control the
color, fill, border, alpha, shape, height, cex, dashline.col, data source of pie if the type is pie.
And also the controls for labels by name the metadata start as
label.parameter.&lt;properties&gt;
such as label.parameter.rot, label.parameter.gp. The parameter is used for
grid.text. The metadata 'featureLayerID' for features are used
for drawing features in different layers. The metadata 'SNPsideID' for SNP.gr
are used for determining the side of lollipops. And the 'SNPsideID' could only
be 'top' or 'bottom'.
</p>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Examples</h3>

<pre><code class="language-R">SNP &lt;- c(10, 100, 105, 108, 400, 410, 420, 600, 700, 805, 840, 1400, 1402)
x &lt;- sample.int(100, length(SNP))
SNP.gr &lt;- GenomicRanges::GRanges("chr1", IRanges::IRanges(SNP, width=1, names=paste0("snp", SNP)),
                                 value1=x, value2=100-x)
SNP.gr$color &lt;- rep(list(c("red", 'blue')), length(SNP))
SNP.gr$border &lt;- sample.int(7, length(SNP), replace=TRUE)
features &lt;- GenomicRanges::GRanges(
  "chr1", IRanges::IRanges(
    c(1, 501, 1001),
    width=c(120, 500, 405),
    names=paste0("block", 1:3)),
  color="black",
  fill=c("#FF8833", "#51C6E6", "#DFA32D"),
  height=c(0.1, 0.05, 0.08),
  label.parameter.rot=45)
lolliplot(SNP.gr, features, type="pie")
</code></pre>


</div>