<div class="container">

<table style="width: 100%;"><tr>
<td>pctopt</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Optimizing cut-off percentage for noise minimization</h2>

<h3>Description</h3>

<p>Finds an optimal cut-off percentage for noise minimization (in vqssub, vqsassess, and vqscustompct functions) that can decrease the number of singleton haplotypes to less than the desired percentage of the total reads.
</p>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fasta</code></td>
<td>
<p>Input as a read alignment in FASTA format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pctsing</code></td>
<td>
<p>The desired percentage of singleton haplotypes relative to the total reads in the alignment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Sequencing error and noise minimization methods that replace low frequency nucleotide base (less than the "pct" cut-off) with consensus base of that position ("conbase": default) or with base of the dominant haplotype ("domhapbase").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samplingfirst</code></td>
<td>
<p>Downsampling before (TRUE) or after (FALSE: default) the noise minimization.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gappct</code></td>
<td>
<p>The percent cut-off particularly specified for gap (-). If it is not specified or less than "pct", "gappct" will be equal to "pct" (default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignoregappositions</code></td>
<td>
<p>Replace all nucleotides in the positions in the alignment containing gap(s) with gap. This will make such positions no longer single nucleotide variant (SNV). The default is "FALSE".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samsize</code></td>
<td>
<p>Sample size (number of reads) after down-sampling. If it is not specified or more than number of reads in the original alignment, down-sampling will not be performed (default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>String within quotation marks indicating name of read alignment (optional).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An optimal cut-off percentage for noise minimization of an input sample and parameter settings. If label is specified, the output will be a data frame with percentage of singleton haplotypes at each cut-off percentage from zero to the optimal cut-off percentage.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Locate input FASTA file-------------------------------------------------------------------------
fastafilepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")

## Find an cut-off percentage that creates singleton haplotypes less than 50% of the alignment.----
pctopt(fastafilepath, pctsing = 50, label = "s1")

</code></pre>


</div>