<div class="container">

<table style="width: 100%;"><tr>
<td>gapremove</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Removing gap-rich positions and/or reads/sequences</h2>

<h3>Description</h3>

<p>Removes nucleotide positions (vertical) and/or reads/sequences (horizontal) that contain gaps more than the specified cut-off percentage from the alignment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gapremove(fasta, vgappct = 70, hgappct = 70, fastaname = "filteredfast.fasta")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fasta</code></td>
<td>
<p>Input as a read or multiple sequence alignment in FASTA format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vgappct</code></td>
<td>
<p>The percent cut-off of vertical gap (-), i.e., if a position in the alignment has %gap &gt;= vgappct, that position will be removed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hgappct</code></td>
<td>
<p>The percent cut-off of horizontal gap (-), i.e., if a sequence or read in the alignment has %gap &gt;= hgappct, that sequence or read will be removed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fastaname</code></td>
<td>
<p>Output file name in FASTA format</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>FASTA read or multiple sequence alignment written out to the input directory
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Locate input FASTA file-------------------------------------------------------------------------
fastafilepath &lt;- system.file("extdata", "gaprichfast.fasta", package = "longreadvqs")

## Indicate output directory and file name---------------------------------------------------------
outfast &lt;- tempfile()

## Remove positions with gap &gt;= 60% and reads/sequences with gap &gt;= 10%----------------------------
gapremove(fastafilepath, vgappct = 60, hgappct = 10, fastaname = outfast)

</code></pre>


</div>