<div class="container">

<table style="width: 100%;"><tr>
<td>tcell</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Microarray Time Series Data for T-Cell Activation </h2>

<h3>Description</h3>

<p>The data result from two experiments investigating
the expression response of human T cells to PMA and ionomicin treatment.
</p>
<p>The first data set (<code>tcell.34</code>) contains the temporal expression levels of 58 genes
for 10 unequally spaced time points.  At each time point there are 34 
separate measurements. The second data set (<code>tcell.10</code>) stems from
a related experiment considering 
the same genes and identical time points, and contains 
10 further measurements per time point. See Rangel et al. (2004) for more details.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(tcell)
</code></pre>


<h3>Format</h3>

<p><code>tcell.10</code> and <code>tcell.34</code> are <code>longitudinal</code> objects,
i.e. matrices with 58 colums each and a number of extra attributes
(see <code>longitudinal</code> and <code>longitudinal.util</code>).
</p>
<p>The vector <code>tcell.gene.descriptions</code> contains the 
description of the functions of the 58 investigated genes.
</p>


<h3>Source</h3>

<p>This data is described in Rangel et al. (2004).
</p>


<h3>References</h3>

<p>Rangel, C., Angus, J., Ghahramani, Z., Lioumi, M., Sotheran, E., Gaiba, A.,
Wild, D. L., and Falciani, F.
(2004) Modeling T-cell activation using gene expression profiling and
state-space models. <em>Bioinformatics</em>, <b>20</b>, 1361â€“1372.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># load "longitudinal" library
library("longitudinal")

# load data sets
data(tcell)

# data set with 10 repeats 
dim(tcell.10)
summary(tcell.10)
is.longitudinal(tcell.10)
is.regularly.sampled(tcell.10)
is.equally.spaced(tcell.10)
get.time.repeats(tcell.10)

# data set with 34 repeats 
dim(tcell.34)
summary(tcell.34)
is.longitudinal(tcell.34)
is.regularly.sampled(tcell.34)
is.equally.spaced(tcell.34)
get.time.repeats(tcell.34)

# descriptions of the first nine genes
tcell.gene.description[1:9]

# plot the first nine time series
plot(tcell.10, 1:9)
plot(tcell.34, 1:9)

# Rangel et al. use the combined data set
tcell.44 &lt;- combine.longitudinal(tcell.34, tcell.10)
plot(tcell.44, 1:9)
</code></pre>


</div>