<div class="container">

<table style="width: 100%;"><tr>
<td>vqscompare</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Comparing viral quasispecies profile and operational taxonomic unit (OTU) classified by k-means clustering between samples</h2>

<h3>Description</h3>

<p>Pools noise-minimized down-sampled read samples and compares their diversity by 1) viral quasispecies profile (haplotype and metrics from QSutils package), 2) operational taxonomic unit (OTU) classified by k-means clustering of single nucleotide variant (SNV) distance, and 3) visualization of different comparative method, i.e., haplotype, OTU, phylogenetic tree, MDS plot. Such comparisons can also be performed at the amino acid level (protein haplotype and single amino acid variation (SAV) group).
</p>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>samplelist</code></td>
<td>
<p>List of samples, i.e., name of resulting objects from "vqsassess" or "vqscustompct" functions, for example list(BC1, BC2, BC3).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lab_name</code></td>
<td>
<p>Name of variable or type of sample for instance "barcode", "sample", "dpi", or "isolate" (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kmeans.n</code></td>
<td>
<p>Number of operational taxonomic units (OTUs) and single amino acid variation (SAV) groups needed from k-means clustering on multidimensional scale (MDS) of all samples' pairwise SNV and SAV distances.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showhap.n</code></td>
<td>
<p>Number of largest haplotypes (default = 30) labeled in the top five OTUs' MDS plot (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>proteincoding</code></td>
<td>
<p>Translate gene or protein-coding reads into amino acid sequences and regroup them into protein haplotypes and single amino acid variation (SAV) groups, which are comparable to haplotypes and OTUs at the nucleotide level, respectively (optional). If not specified or if proteincoding = FALSE, gene translation and downstream analyses will not be performed (default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>removestopcodon</code></td>
<td>
<p>Remove the last amino acid (expected to be a stop codon) from translated amino acid sequences before further analysis (optional). If not specified or if removestopcodon = FALSE, the last amino acid will not be removed (default).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List of 1) "hapdiv": comparative table of viral quasispecies diversity metrics between listed samples calculated by QSutils package, 2) "otudiv": comparative table of operational taxonomic unit (OTU) diversity metrics between listed samples calculated from consensus sequence of each OTU (similar to "otucompare" function's output), 3) "sumsnv_hap_otu": frequency and SNV profile (by position in the alignment) of all haplotypes and OTUs, 4) "fullseq": complete read sequence of all haplotypes, 5) "fulldata": complete read sequence of all haplotypes in every sample with frequency and OTU classification, 6) "summaryplot": visualization of viral quasispecies comparison between samples including  6.1) "happlot": proportion of haplotypes (top left), 6.2) "otuplot": proportion of OTUs (bottom left), 6.3) multidimensional scale (MDS) plots (right) of k-means OTU ("top5otumds": 5 largest groups with major haplotypes labeled and "allotumds": all groups), 7) "aadiv": comparative table of viral quasispecies diversity metrics between listed samples based on translated reads calculated by QSutils package (similar to one of "AAcompare" function's outputs), 8) "savgrpdiv": comparative table of single amino acid (SAV) group diversity metrics between listed samples calculated from consensus amino acid sequence of each SAV group (similar to one of "AAcompare" function's outputs), 9) "sumsav_phap_savgrp": frequency and SAV profile (by position in the alignment) of all protein haplotypes and SAV groups, 10) "fullseq_aa": complete amino acid sequence of all protein haplotypes, 11) "fulldata_aa": complete amino acid sequence of all protein haplotypes in every sample with frequency and SAV group classification, 12) "summaryplot_aa": visualization of viral quasispecies comparison between samples based on translated reads including  12.1) "phapplot": proportion of protein haplotypes (top left), 12.2) "savgrpplot": proportion of SAV groups (bottom left), 12.3) multidimensional scale (MDS) plots (right) of k-means SAV group ("top5savgrpmds": 5 largest SAV groups with major protein haplotypes labeled and "allsavgrpmds": all SAV groups), 7) to 12) will be generated only when proteincoding = TRUE
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Locate input FASTA files-----------------------------------------------------------------------
sample1filepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")
sample2filepath &lt;- system.file("extdata", "s2.fasta", package = "longreadvqs")

## Prepare data for viral quasispecies comparison between two samples-----------------------------
set.seed(123)
sample1 &lt;- vqsassess(sample1filepath, pct = 5, samsize = 50, label = "sample1")
sample2 &lt;- vqsassess(sample2filepath, pct = 5, samsize = 50, label = "sample2")

## Compare viral quasispecies and OTU (4 clusters) diversity between two samples------------------
out &lt;- vqscompare(samplelist = list(sample1, sample2),
           lab_name = "Sample", kmeans.n = 4, showhap.n = 5)
out$summaryplot

</code></pre>


</div>