<div class="container">

<table style="width: 100%;"><tr>
<td>vqsout</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Exporting viral quasispecies profile comparison results</h2>

<h3>Description</h3>

<p>Writes out resulting objects from "vqscompare" function as tables (TSV files) and alignment (FASTA file) to the working directory.
</p>


<h3>Usage</h3>

<pre><code class="language-R">vqsout(vqscompare.obj, directory = "path/to/directory")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vqscompare.obj</code></td>
<td>
<p>A resulting object from "vqscompare" function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>directory</code></td>
<td>
<p>Path to desired directory (location) for output files. If it is not specified, the directory will be the current working directory.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>TSV files of viral quasispecies profile comparison results and FASTA file of unique haplotype alignment.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Locate input FASTA files-----------------------------------------------------------------------
sample1filepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")
sample2filepath &lt;- system.file("extdata", "s2.fasta", package = "longreadvqs")

## Prepare data for viral quasispecies comparison between two samples-----------------------------
set.seed(123)
sample1 &lt;- vqsassess(sample1filepath, pct = 0, samsize = 50, label = "sample1")
sample2 &lt;- vqsassess(sample2filepath, pct = 0, samsize = 50, label = "sample2")

## Compare viral quasispecies and OTU (4 clusters) diversity between two samples------------------
comp &lt;- vqscompare(samplelist = list(sample1, sample2),
           lab_name = "Sample", kmeans.n = 4, showhap.n = 5)

## Export Key outputs from "vqscompare" function--------------------------------------------------
notrun &lt;- vqsout(comp, directory = tempdir())

</code></pre>


</div>