<div class="container">

<table style="width: 100%;"><tr>
<td>snvcompare</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plotting single nucleotide variant (SNV) frequency in read alignment across different samples</h2>

<h3>Description</h3>

<p>Compares single nucleotide variant (SNV) profile between noise-minimized down-sampled read samples using cowplot's "plot_grid" function. The resulting plot may help evaluating the optimal cut-off percentage of low frequency nucleotide base used in "vqsassess", "vqscustompct", or "vqssub" functions.
</p>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>samplelist</code></td>
<td>
<p>List of samples, i.e., name of resulting objects from "vqsassess" or "vqscustompct" functions, for example list(BC1, BC2, BC3).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncol</code></td>
<td>
<p>Number of columns for multiple plots (see cowplot's "plot_grid" function)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Comparative plot of SNV frequency in read alignment across different samples
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Locate input FASTA files-----------------------------------------------------------------------
sample1filepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")
sample2filepath &lt;- system.file("extdata", "s2.fasta", package = "longreadvqs")

## Prepare data for viral quasispecies comparison between two samples-----------------------------
sample1 &lt;- vqsassess(sample1filepath, pct = 10, label = "sample1")
sample2 &lt;- vqsassess(sample2filepath, pct = 10, label = "sample2")

## Compare SNV profile between two listed samples-------------------------------------------------
snvcompare(samplelist = list(sample1, sample2), ncol = 1)

</code></pre>


</div>