<div class="container">

<table style="width: 100%;"><tr>
<td>Filter_function</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Filter OTU/ASV/metagenomic profile/gene profile by threshold</h2>

<h3>Description</h3>

<p>Sequenced data of taxonomy&amp;gene still remains some sequencing error which we needed to be wiped off before analyzing.
Here we provide function including four formats to wipe them clean.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Filter_function(input, threshold, format, report = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>
<p>Data frame of absolute abundance of standard OTU table,with the first column of OTUID and the final column of taxonomy annotation.
If your data frame is gene table or not a standard OTU table, please manually transformed into a standard input data frame.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>threshold of filter.Relative abundance for format 1 and 4, reads number for format 2, sample size for format 3</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>1:filter OTU/gene below overall-sample relative abundance threshold(&lt;)
</p>
<p>2:filter OTU/gene below overall-sample reads threshold(&lt;)
</p>
<p>3:filter OTU/gene reads 0 over threshold sample size(&gt;)
</p>
<p>4:filter OTU/gene below relative abundance threshold in each sample(&lt;)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>report</code></td>
<td>
<p>Logical. If print report to console. Default:TRUE</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Dataframe of OTU/gene in format of absolute abundacne(reads) or relative abundance(%)
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi
</p>


<h3>Examples</h3>

<pre><code class="language-R">### Data frame with absolute abundance (reads)###
### And first column of OTUID and last column of taxonomy ###
data(testotu)

#### If your data frame does not contain the OTUID column or taxonomy column,
#### you can add a simulated column to fit the input format like testotu ##

### 1. Filter OTU with total relative abundance below 0.0001###
filtered_otu &lt;- Filter_function(
  input = testotu,
  threshold = 0.0001,
  format = 1
)

### 2. Filter OTU with total reads below 20 ###
filtered_otu &lt;- Filter_function(
  input = testotu,
  threshold = 20,
  format = 2
)

### 3. Filter OTU reads 0 over (&gt;=) 11 samples ###
filtered_otu &lt;- Filter_function(
  input = testotu,
  threshold = 11,
  format = 3
)

### 4. Filter OTU with relative abundance below 0.0001 in each sample ###
filtered_otu &lt;- Filter_function(
  input = testotu,
  threshold = 0.0001,
  format = 4
)
</code></pre>


</div>